"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[95],{3905:(e,t,a)=>{a.d(t,{Zo:()=>u,kt:()=>m});var n=a(67294);function i(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){i(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function s(e,t){if(null==e)return{};var a,n,i=function(e,t){if(null==e)return{};var a,n,i={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(i[a]=e[a]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var l=n.createContext({}),p=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},u=function(e){var t=p(e.components);return n.createElement(l.Provider,{value:t},e.children)},d="mdxType",c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},g=n.forwardRef((function(e,t){var a=e.components,i=e.mdxType,r=e.originalType,l=e.parentName,u=s(e,["components","mdxType","originalType","parentName"]),d=p(a),g=i,m=d["".concat(l,".").concat(g)]||d[g]||c[g]||r;return a?n.createElement(m,o(o({ref:t},u),{},{components:a})):n.createElement(m,o({ref:t},u))}));function m(e,t){var a=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=a.length,o=new Array(r);o[0]=g;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[d]="string"==typeof e?e:i,o[1]=s;for(var p=2;p<r;p++)o[p]=a[p];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}g.displayName="MDXCreateElement"},36965:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>c,frontMatter:()=>r,metadata:()=>s,toc:()=>p});var n=a(87462),i=(a(67294),a(3905));const r={sidebar_position:1},o="Getting the data",s={unversionedId:"genome_wide_association_studies/quantitative_trait_gwas/getting_the_data",id:"genome_wide_association_studies/quantitative_trait_gwas/getting_the_data",title:"Getting the data",description:"You will be using four sets of files for this exercise. It is recommended that you download these files to your working directory prior to starting the exercise Genotype data files.  To get started, make a new directory to run the tutorial in and cd into it:",source:"@site/docs/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data.md",sourceDirName:"genome_wide_association_studies/quantitative_trait_gwas",slug:"/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"sidebar8",previous:{title:"Genome-wide association with quantitative antibody levels",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/"},next:{title:"Getting the software",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software"}},l={},p=[{value:"What&#39;s in the data?",id:"whats-in-the-data",level:2},{value:"Next steps",id:"next-steps",level:2}],u={toc:p},d="wrapper";function c(e){let{components:t,...a}=e;return(0,i.kt)(d,(0,n.Z)({},u,a,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"getting-the-data"},"Getting the data"),(0,i.kt)("p",null,"You will be using four sets of files for this exercise. It is recommended that you download these files to your working directory prior to starting the exercise Genotype data files.  To get started, make a new directory to run the tutorial in and ",(0,i.kt)("inlineCode",{parentName:"p"},"cd")," into it:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"mkdir gwas_tutorial\ncd gwas_tutorial\n")),(0,i.kt)("p",null,"Now download the data..."),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"curl -O https://www.chg.ox.ac.uk/bioinformatics/training/gms/data/gwas/quantitative_trait_gwas.tgz\n")),(0,i.kt)("p",null,"...and unpack it:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"tar -xzf quantitative_trait_gwas.tgz\n")),(0,i.kt)("p",null,"Once the data is unpacked, you can safely delete the ",(0,i.kt)("inlineCode",{parentName:"p"},"quantitative_trait_gwas.tgz")," file to save space, if you like."),(0,i.kt)("p",null,"Finally, to get started, we'll assume you are working within the ",(0,i.kt)("inlineCode",{parentName:"p"},"gwas_tutorial")," folder:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"cd gwas_tutorial\n")),(0,i.kt)("h2",{id:"whats-in-the-data"},"What's in the data?"),(0,i.kt)("p",null,"You should now have four folders which have this structure:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"gwas_tutorial/\n    Genotype_data/\n        AMR_genotypes.bed\n        ...\n    Imputed_data/\n        AMR_imputed.gen.gz\n        ...\n    Phenotype_data/\n        AMR_phenotype.txt\n        ...\n    Example_scripts/\n        ...\n")),(0,i.kt)("p",null,"The data is as follows:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"The ",(0,i.kt)("inlineCode",{parentName:"p"},"Genotype_data/")," folder contains ",(0,i.kt)("strong",{parentName:"p"},"sample genotypes"),", as typed on a ",(0,i.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/DNA_microarray"},"DNA microarray"),".  Take a look - you should see three files - a ",(0,i.kt)("inlineCode",{parentName:"p"},".bed")," file, a ",(0,i.kt)("inlineCode",{parentName:"p"},".bim")," file, and ",(0,i.kt)("inlineCode",{parentName:"p"}," .fam")," file.  (These are in ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/input#bed"},"plink binary format"),", which is a commonly-used file format for GWAS data.)")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"The ",(0,i.kt)("inlineCode",{parentName:"p"},"Imputed_data")," folder contains ",(0,i.kt)("strong",{parentName:"p"},"imputed genotype files"),", which is the same data 'imputed' up to the full 1000 Genomes Phase 3 reference panel.  These are like the array genotype files but at a much higher resolution (more genetic variants), having been filled in using ",(0,i.kt)("a",{parentName:"p",href:"https://doi.org/10.1038/s10038-023-01213-6"},"genotype imputation"),".")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"The ",(0,i.kt)("inlineCode",{parentName:"p"},"Phenotype_data")," folder contains (guess what?) ",(0,i.kt)("strong",{parentName:"p"},"phenotype information"),".  There are two files: one reports the measured norovirus antibody response for each sample, and the other reflects the ",(0,i.kt)("strong",{parentName:"p"},"secretor status"),".")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"There are also some example scripts are in the ",(0,i.kt)("inlineCode",{parentName:"p"},"Example_scripts")," directory."))),(0,i.kt)("h2",{id:"next-steps"},"Next steps"),(0,i.kt)("p",null,"As with all GWAS, the first place to start is to summarise and perform ",(0,i.kt)("a",{parentName:"p",href:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes"},"quality control of the data"),"."))}c.isMDXComponent=!0}}]);