"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[4671],{3905:(e,t,a)=>{a.d(t,{Zo:()=>m,kt:()=>u});var n=a(67294);function i(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){i(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,i=function(e,t){if(null==e)return{};var a,n,i={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(i[a]=e[a]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var s=n.createContext({}),d=function(e){var t=n.useContext(s),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},m=function(e){var t=d(e.components);return n.createElement(s.Provider,{value:t},e.children)},p="mdxType",c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},f=n.forwardRef((function(e,t){var a=e.components,i=e.mdxType,r=e.originalType,s=e.parentName,m=l(e,["components","mdxType","originalType","parentName"]),p=d(a),f=i,u=p["".concat(s,".").concat(f)]||p[f]||c[f]||r;return a?n.createElement(u,o(o({ref:t},m),{},{components:a})):n.createElement(u,o({ref:t},m))}));function u(e,t){var a=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=a.length,o=new Array(r);o[0]=f;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l[p]="string"==typeof e?e:i,o[1]=l;for(var d=2;d<r;d++)o[d]=a[d];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}f.displayName="MDXCreateElement"},33341:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>s,contentTitle:()=>o,default:()=>c,frontMatter:()=>r,metadata:()=>l,toc:()=>d});var n=a(87462),i=(a(67294),a(3905));const r={sidebar_position:1},o="PLINK file formats",l={unversionedId:"genome_wide_association_studies/file_formats/plink",id:"genome_wide_association_studies/file_formats/plink",title:"PLINK file formats",description:"Data in binary PLINK format, such as that provided for this tutorial, consists of a set of three files each containing information about the genotyping data and the samples included in the dataset. A brief description of each is provided below. More information on file formats can be found on the PLINK website",source:"@site/docs/genome_wide_association_studies/file_formats/plink.md",sourceDirName:"genome_wide_association_studies/file_formats",slug:"/genome_wide_association_studies/file_formats/plink",permalink:"/chg-training-resources/genome_wide_association_studies/file_formats/plink",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/file_formats/plink.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"sidebar8",previous:{title:"GWAS file formats",permalink:"/chg-training-resources/genome_wide_association_studies/file_formats/"},next:{title:"GCTA file formats",permalink:"/chg-training-resources/genome_wide_association_studies/file_formats/gcta"}},s={},d=[{value:".bed files",id:"bed-files",level:3},{value:".bim files",id:"bim-files",level:3},{value:".fam files",id:"fam-files",level:3},{value:".qassoc files",id:"qassoc-files",level:3}],m={toc:d},p="wrapper";function c(e){let{components:t,...a}=e;return(0,i.kt)(p,(0,n.Z)({},m,a,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"plink-file-formats"},"PLINK file formats"),(0,i.kt)("p",null,"Data in binary PLINK format, such as that provided for this tutorial, consists of a set of three files each containing information about the genotyping data and the samples included in the dataset. A brief description of each is provided below. More information on file formats can be found on the PLINK website"),(0,i.kt)("h3",{id:"bed-files"},".bed files"),(0,i.kt)("p",null,"The binary .bed file contains the genotype calls for each individual in the dataset. As a binary file, data is encoded as bytes, and not easily readable except through programs such as PLINK."),(0,i.kt)("h3",{id:"bim-files"},".bim files"),(0,i.kt)("p",null,"These text files contain information about each variant in the dataset. The file consists of one line per variant, with the following columns:"),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},"Chromosome name"),(0,i.kt)("li",{parentName:"ol"},"Variant identifier"),(0,i.kt)("li",{parentName:"ol"},"Genomic position in morgans or centimorgans"),(0,i.kt)("li",{parentName:"ol"},"Base-pair position (1-based)"),(0,i.kt)("li",{parentName:"ol"},"Allele 1 (usually the minor allele)"),(0,i.kt)("li",{parentName:"ol"},"Allele 2 (usually the major allele)")),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"PLINK currently does not handle multi-alellic SNPs (those with more than one alternate allele)"),(0,i.kt)("p",{parentName:"div"},"It also has a bad habit of 'flipping' data so the first allele is the minor allele (in the data). If you want to avoid\nthat, you can run all the commands with the ",(0,i.kt)("inlineCode",{parentName:"p"},"--keep-allele-order")," option, but we've skipped that in this tutorial."))),(0,i.kt)("h3",{id:"fam-files"},".fam files"),(0,i.kt)("p",null,"These contain information about each sample in the dataset. The file consists of one line per sample, with the following columns:"),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},"Family ID"),(0,i.kt)("li",{parentName:"ol"},"Within-family ID (Individual ID)"),(0,i.kt)("li",{parentName:"ol"},'Paternal ID (or "0" if not in dataset)'),(0,i.kt)("li",{parentName:"ol"},'Maternal ID (or "0" if not in dataset)'),(0,i.kt)("li",{parentName:"ol"},'Sex ("1" = male, "2" = female, "0" = unknown)'),(0,i.kt)("li",{parentName:"ol"},'Phenotype value ("-9" = missing; in this data, phenotypes are provided in a separate file)')),(0,i.kt)("h3",{id:"qassoc-files"},".qassoc files"),(0,i.kt)("p",null,"These contain the summary statistics for each SNP included in the association analysis. The file has a header line followed by a set of statistics for each SNP. The column names and contents are as follows:"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"column"),(0,i.kt)("th",{parentName:"tr",align:null},"description"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"CHR")),(0,i.kt)("td",{parentName:"tr",align:null},"Chromosome code")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"SNP")),(0,i.kt)("td",{parentName:"tr",align:null},"Variant identifier")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"BP")),(0,i.kt)("td",{parentName:"tr",align:null},"Base-pair coordinate")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"NMISS")),(0,i.kt)("td",{parentName:"tr",align:null},"Number of nonmissing genotype calls")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"BETA")),(0,i.kt)("td",{parentName:"tr",align:null},"Regression coefficient")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"SE")),(0,i.kt)("td",{parentName:"tr",align:null},"Standard error")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"R2")),(0,i.kt)("td",{parentName:"tr",align:null},"Regression r-squared")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"T")),(0,i.kt)("td",{parentName:"tr",align:null},"Wald test (based on t-distribution)")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"P")),(0,i.kt)("td",{parentName:"tr",align:null},"Wald test asymptotic p-value")))))}c.isMDXComponent=!0}}]);