"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[3605],{3905:(e,t,n)=>{n.d(t,{Zo:()=>d,kt:()=>u});var a=n(67294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function l(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},r=Object.keys(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var s=a.createContext({}),p=function(e){var t=a.useContext(s),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},d=function(e){var t=p(e.components);return a.createElement(s.Provider,{value:t},e.children)},m="mdxType",f={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},c=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,r=e.originalType,s=e.parentName,d=l(e,["components","mdxType","originalType","parentName"]),m=p(n),c=i,u=m["".concat(s,".").concat(c)]||m[c]||f[c]||r;return n?a.createElement(u,o(o({ref:t},d),{},{components:n})):a.createElement(u,o({ref:t},d))}));function u(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=n.length,o=new Array(r);o[0]=c;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l[m]="string"==typeof e?e:i,o[1]=l;for(var p=2;p<r;p++)o[p]=n[p];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}c.displayName="MDXCreateElement"},77083:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>s,contentTitle:()=>o,default:()=>f,frontMatter:()=>r,metadata:()=>l,toc:()=>p});var a=n(87462),i=(n(67294),n(3905));const r={sidebar_position:3},o="'Oxford' file formats format",l={unversionedId:"genome_wide_association_studies/file_formats/oxford",id:"genome_wide_association_studies/file_formats/oxford",title:"'Oxford' file formats format",description:".gen file",source:"@site/docs/genome_wide_association_studies/file_formats/oxford.md",sourceDirName:"genome_wide_association_studies/file_formats",slug:"/genome_wide_association_studies/file_formats/oxford",permalink:"/chg-training-resources/genome_wide_association_studies/file_formats/oxford",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/file_formats/oxford.md",tags:[],version:"current",sidebarPosition:3,frontMatter:{sidebar_position:3},sidebar:"sidebar8",previous:{title:"GCTA file formats",permalink:"/chg-training-resources/genome_wide_association_studies/file_formats/gcta"}},s={},p=[{value:".gen file",id:"gen-file",level:3},{value:".sample file",id:"sample-file",level:3},{value:".info file",id:"info-file",level:3}],d={toc:p},m="wrapper";function f(e){let{components:t,...n}=e;return(0,i.kt)(m,(0,a.Z)({},d,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"oxford-file-formats-format"},"'Oxford' file formats format"),(0,i.kt)("h3",{id:"gen-file"},".gen file"),(0,i.kt)("p",null,"The GEN file format is designed to hold either directly-typed or imputed genotype data.\nTo handle this it stores probabilities of each genotype for each sample, rather than a hard-called genotype."),(0,i.kt)("p",null,"The file consists of one line per SNP, with the following columns:"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"column"),(0,i.kt)("th",{parentName:"tr",align:null},"description"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"chromosome")),(0,i.kt)("td",{parentName:"tr",align:null},"The chromosome identifier. ",(0,i.kt)("strong",{parentName:"td"},"Note.")," this column is optional - not all GEN files have it.")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"SNPID")),(0,i.kt)("td",{parentName:"tr",align:null},"An ID for the SNP")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"rsid")),(0,i.kt)("td",{parentName:"tr",align:null},"Another ID for the SNP")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"position")),(0,i.kt)("td",{parentName:"tr",align:null},"Position")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"Allele A")),(0,i.kt)("td",{parentName:"tr",align:null},"The first allele in the data")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"Allele B")),(0,i.kt)("td",{parentName:"tr",align:null},"The second allele in the data")))),(0,i.kt)("p",null,"The remaining columns are in sets of 3 containing genotype probabilities for genotypes ",(0,i.kt)("inlineCode",{parentName:"p"},"AA"),", ",(0,i.kt)("inlineCode",{parentName:"p"},"AB"),", and ",(0,i.kt)("inlineCode",{parentName:"p"},"BB")," for each sample."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"See also the ",(0,i.kt)("a",{parentName:"p",href:"https://www.bgenformat.org"},"BGEN format")," which is widely used for imputed genotype data."))),(0,i.kt)("h3",{id:"sample-file"},".sample file"),(0,i.kt)("p",null,"The sample file has three parts"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"a header line detailing the names of the columns in the file"),(0,i.kt)("li",{parentName:"ul"},"a line detailing the types of variables stored in each column - this is either '0' (for identifier), 'D' (discrete variable), 'B' for a binary phenotype, or 'C' or 'P' for continuous variables."),(0,i.kt)("li",{parentName:"ul"},"a line for each individual detailing the information for that individual")),(0,i.kt)("p",null,"Here's an example of a ",(0,i.kt)("inlineCode",{parentName:"p"},".sample")," file:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"ID pheno sex cov1 cov2\n0 B D D C\nsample_1 1 male british 2.5\nsample_2 0 female british 12.2\nsample_2 0 female french 12.2\n...\n")),(0,i.kt)("h3",{id:"info-file"},".info file"),(0,i.kt)("p",null,"This file consists of one line per SNP and a single header line at the beginning. This file always contains the following columns (header tags shown in parentheses):"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"SNP identifier from -g file (snp_id)"),(0,i.kt)("li",{parentName:"ul"},"rsID (rs_id)"),(0,i.kt)("li",{parentName:"ul"},"base pair position (position)"),(0,i.kt)("li",{parentName:"ul"},"expected frequency of allele coded '1' in the -o file (exp_freq_a1)"),(0,i.kt)("li",{parentName:"ul"},"measure of the observed statistical information associated with the allele frequency estimate (info)"),(0,i.kt)("li",{parentName:"ul"},"average certainty of best-guess genotypes (certainty)"),(0,i.kt)("li",{parentName:"ul"},'internal "type" assigned to SNP (type)')))}f.isMDXComponent=!0}}]);