"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[9627],{3905:(e,a,t)=>{t.d(a,{Zo:()=>l,kt:()=>d});var n=t(67294);function s(e,a,t){return a in e?Object.defineProperty(e,a,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[a]=t,e}function r(e,a){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);a&&(n=n.filter((function(a){return Object.getOwnPropertyDescriptor(e,a).enumerable}))),t.push.apply(t,n)}return t}function m(e){for(var a=1;a<arguments.length;a++){var t=null!=arguments[a]?arguments[a]:{};a%2?r(Object(t),!0).forEach((function(a){s(e,a,t[a])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):r(Object(t)).forEach((function(a){Object.defineProperty(e,a,Object.getOwnPropertyDescriptor(t,a))}))}return e}function p(e,a){if(null==e)return{};var t,n,s=function(e,a){if(null==e)return{};var t,n,s={},r=Object.keys(e);for(n=0;n<r.length;n++)t=r[n],a.indexOf(t)>=0||(s[t]=e[t]);return s}(e,a);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)t=r[n],a.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(s[t]=e[t])}return s}var i=n.createContext({}),o=function(e){var a=n.useContext(i),t=a;return e&&(t="function"==typeof e?e(a):m(m({},a),e)),t},l=function(e){var a=o(e.components);return n.createElement(i.Provider,{value:a},e.children)},c="mdxType",N={inlineCode:"code",wrapper:function(e){var a=e.children;return n.createElement(n.Fragment,{},a)}},h=n.forwardRef((function(e,a){var t=e.components,s=e.mdxType,r=e.originalType,i=e.parentName,l=p(e,["components","mdxType","originalType","parentName"]),c=o(t),h=s,d=c["".concat(i,".").concat(h)]||c[h]||N[h]||r;return t?n.createElement(d,m(m({ref:a},l),{},{components:t})):n.createElement(d,m({ref:a},l))}));function d(e,a){var t=arguments,s=a&&a.mdxType;if("string"==typeof e||s){var r=t.length,m=new Array(r);m[0]=h;var p={};for(var i in a)hasOwnProperty.call(a,i)&&(p[i]=a[i]);p.originalType=e,p[c]="string"==typeof e?e:s,m[1]=p;for(var o=2;o<r;o++)m[o]=t[o];return n.createElement.apply(null,m)}return n.createElement.apply(null,t)}h.displayName="MDXCreateElement"},26263:(e,a,t)=>{t.r(a),t.d(a,{assets:()=>i,contentTitle:()=>m,default:()=>N,frontMatter:()=>r,metadata:()=>p,toc:()=>o});var n=t(87462),s=(t(67294),t(3905));const r={sidebar_position:101},m="A bayesian forest plot",p={unversionedId:"statistical_modelling/introduction/allele_frequencies_2",id:"statistical_modelling/introduction/allele_frequencies_2",title:"A bayesian forest plot",description:"The plot we just made is ok, but surely visually pretty wasteful.",source:"@site/docs/statistical_modelling/introduction/allele_frequencies_2.md",sourceDirName:"statistical_modelling/introduction",slug:"/statistical_modelling/introduction/allele_frequencies_2",permalink:"/chg-training-resources/statistical_modelling/introduction/allele_frequencies_2",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/statistical_modelling/introduction/allele_frequencies_2.md",tags:[],version:"current",sidebarPosition:101,frontMatter:{sidebar_position:101},sidebar:"sidebar6",previous:{title:"Estimating allele frequencies",permalink:"/chg-training-resources/statistical_modelling/introduction/allele_frequencies"},next:{title:"The normal approximation",permalink:"/chg-training-resources/statistical_modelling/introduction/challenge"}},i={},o=[{value:"Making a forest plot",id:"making-a-forest-plot",level:2},{value:"Forest plot &#39;howto&#39;",id:"forest-plot-howto",level:2}],l={toc:o},c="wrapper";function N(e){let{components:a,...r}=e;return(0,s.kt)(c,(0,n.Z)({},l,r,{components:a,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"a-bayesian-forest-plot"},"A bayesian forest plot"),(0,s.kt)("p",null,"The plot ",(0,s.kt)("a",{parentName:"p",href:"/chg-training-resources/statistical_modelling/introduction/allele_frequencies"},"we just made")," is ok, but surely visually pretty wasteful."),(0,s.kt)("p",null,"For most purposes a better alternative is ",(0,s.kt)("strong",{parentName:"p"},"forest plot")," - that is a plot showing ",(0,s.kt)("em",{parentName:"p"},"point estimates")," and their ",(0,s.kt)("em",{parentName:"p"},"95% credible intervals"),"."),(0,s.kt)("h2",{id:"making-a-forest-plot"},"Making a forest plot"),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Create a ",(0,s.kt)("strong",{parentName:"p"},"forest plot")," for these estimates, that is, a plot of the posterior mean estimates and the 95% intervals.\nSchematically it should look schematically something like this:"),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("img",{alt:"img",src:t(76427).Z,width:"1508",height:"678"})),(0,s.kt)("p",{parentName:"div"},"For ",(0,s.kt)("strong",{parentName:"p"},"extra kudos"),", make a version with a prior (e.g. ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msub",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msub"},"n"),(0,s.kt)("mi",{parentName:"msub"},"A")),(0,s.kt)("mo",{parentName:"mrow"},"="),(0,s.kt)("msub",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msub"},"n"),(0,s.kt)("mi",{parentName:"msub"},"B")),(0,s.kt)("mo",{parentName:"mrow"},"="),(0,s.kt)("mn",{parentName:"mrow"},"5")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"n_A = n_B = 5")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.5806em",verticalAlign:"-0.15em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"n"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3283em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"0em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"A")))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,s.kt)("span",{parentName:"span"})))))),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},"="),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.5806em",verticalAlign:"-0.15em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"n"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3283em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"0em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal mtight",style:{marginRight:"0.05017em"}},"B")))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,s.kt)("span",{parentName:"span"})))))),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},"="),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"5"))))),", for example, or a prior of your choice) and compare to the without-prior version."))),(0,s.kt)("h2",{id:"forest-plot-howto"},"Forest plot 'howto'"),(0,s.kt)("p",null,"In case you're stuck here is a guide."),(0,s.kt)("p",null,"First, instead of that function ",(0,s.kt)("inlineCode",{parentName:"p"},"compute_posterior_density()"),", write a simpler function ",(0,s.kt)("inlineCode",{parentName:"p"},"compute_posterior_summary()")," that summarises the posterior, like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},"compute_posterior_summary = function(\n    nA, nB,\n    prior = c( nA = 0, nB = 0 )\n) {\n    return(\n        data.frame(\n            posterior_mean = # your calculation here\n            q2.5 = # your calculation here\n            q97.5 = # your calculation here\n        )\n    )\n}\n")),(0,s.kt)("p",null,"As indicated, it should return the data we need for the forest plot. (Returning this as a dataframe with one row works best).\nThe columns should be the posterior ",(0,s.kt)("strong",{parentName:"p"},"mean")," and - to find an interval containing 95% of the mass - the ",(0,s.kt)("strong",{parentName:"p"},"2.5%")," and ",(0,s.kt)("strong",{parentName:"p"},"97.5% quantiles"),".   (Feel free to add other quantiles as well - for example the 50% quantile, which is also known as the ",(0,s.kt)("strong",{parentName:"p"},"median"),".)"),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"If you're working in R, you can compute the quantiles using the ",(0,s.kt)("inlineCode",{parentName:"p"},"qbeta()")," function.  For example to get the 2.5% quantile:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},"qbeta( p = 0.025, shape1 = nB+1, shape2 = nA+1 )\n")),(0,s.kt)("p",{parentName:"div"},"(though you should also add the prior information in, of course!)"))),(0,s.kt)("p",null,"Happily, it turns out that the posterior mean is just the frequency estimate you would have made from the data\nanyway:"),(0,s.kt)("div",{className:"math math-display"},(0,s.kt)("span",{parentName:"div",className:"katex-display"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mtext",{parentName:"mrow"},"posterior\xa0mean"),(0,s.kt)("mo",{parentName:"mrow"},"="),(0,s.kt)("mfrac",{parentName:"mrow"},(0,s.kt)("msub",{parentName:"mfrac"},(0,s.kt)("mi",{parentName:"msub"},"n"),(0,s.kt)("mi",{parentName:"msub"},"B")),(0,s.kt)("mrow",{parentName:"mfrac"},(0,s.kt)("msub",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msub"},"n"),(0,s.kt)("mi",{parentName:"msub"},"A")),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("msub",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msub"},"n"),(0,s.kt)("mi",{parentName:"msub"},"B"))))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\text{posterior mean} = \\frac{n_B}{n_A+n_B}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8623em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"posterior\xa0mean")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},"="),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"1.9436em",verticalAlign:"-0.836em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,s.kt)("span",{parentName:"span",className:"mfrac"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.1076em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.314em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"n"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3283em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"0em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"A")))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,s.kt)("span",{parentName:"span"})))))),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"n"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3283em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"0em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal mtight",style:{marginRight:"0.05017em"}},"B")))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,s.kt)("span",{parentName:"span"})))))))),(0,s.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,s.kt)("span",{parentName:"span",style:{top:"-3.677em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"n"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3283em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"0em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal mtight",style:{marginRight:"0.05017em"}},"B")))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,s.kt)("span",{parentName:"span"}))))))))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.836em"}},(0,s.kt)("span",{parentName:"span"}))))),(0,s.kt)("span",{parentName:"span",className:"mclose nulldelimiter"}))))))),(0,s.kt)("p",null,"...so that's easy to compute as well.  "),(0,s.kt)("p",null,"Once you've written your function, you could use it like this to compute a summary:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},"posterior_summary = (\n    data\n    %>% mutate(\n        nA = ( `C/C` + `-/C` ),\n        nB = ( `-/-` )\n    )\n    %>% group_by( population )\n    %>% summarise( compute_posterior_summary( nA, nB, prior = c( nA = 0, nB = 0 )))\n)\n")),(0,s.kt)("p",null,"You should end up with a data frame looking something like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"population            n1    n2   posterior_mean   q2.5   q97.5\nAfrican Ancestry SW   32    29            0.475  0.355   0.599\n(etc).\n")),(0,s.kt)("p",null,"Now let's change the plotting code so it plots a forest plot.\nSee the comments below for an explanation"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},'p = (\n    ggplot( data = posterior_summary )\n    + geom_point( aes( x = posterior_mean, y = population ))\n    + geom_segment( aes( x = q2.5, xend = q97.5, y = population, yend = population ))\n    # Always make your plot have good axis labels!\n    + xlab( "Posterior mean and 95% CI" )\n    + ylab( "Population" )\n    # A clean, large-font theme:\n    + theme_minimal(16)\n    # Rotate y axis label so it\'s not at 90 degrees:\n    + theme( axis.title.y = element_text( angle = 0, vjust = 0.5 ))\n)\nprint(p)\n')),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Note.")," If the points are not in order, you can ",(0,s.kt)("a",{parentName:"p",href:"/chg-training-resources/statistical_modelling/introduction/allele_frequencies#ordering-populations"},"order them in the data")," before re-running the above."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Which plot do you like better - the ",(0,s.kt)("a",{parentName:"p",href:"/chg-training-resources/statistical_modelling/introduction/allele_frequencies"},"full posterior")," or this one?"),(0,s.kt)("p",{parentName:"div"},"Do the 95% intervals all overlap?"))),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Comparing the inference")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Compare the bayesian shinkage and non-shrinkage estimates by plotting different values side-by side.\nHere is an easy way to do this:"),(0,s.kt)("ol",{parentName:"div"},(0,s.kt)("li",{parentName:"ol"},"Update the above code to add an extra column called ",(0,s.kt)("inlineCode",{parentName:"li"},"prior")," to the ",(0,s.kt)("inlineCode",{parentName:"li"},"posterior_summary")," data frame. The above had no prior data, so give it the value 'none'."),(0,s.kt)("li",{parentName:"ol"},"Now add ",(0,s.kt)("em",{parentName:"li"},"extra rows")," to ",(0,s.kt)("inlineCode",{parentName:"li"},"posterior_summary")," for a chosen different prior - say ",(0,s.kt)("inlineCode",{parentName:"li"},"nA=5"),", ",(0,s.kt)("inlineCode",{parentName:"li"},"nB=5")," or other values of your choice.  (Make sure to change the ",(0,s.kt)("inlineCode",{parentName:"li"},"prior")," column to something different, like '5,5'!)")),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Note"),". the ",(0,s.kt)("inlineCode",{parentName:"p"},"rbind()")," function in R (as well as ",(0,s.kt)("inlineCode",{parentName:"p"},"dplyr::bind_rows()"),") will concatenate two data frames, row-wise."),(0,s.kt)("ol",{parentName:"div",start:3},(0,s.kt)("li",{parentName:"ol"},"Lastly, try re-making your plot but adding")),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"+ facet_grid( . ~ prior )\n")),(0,s.kt)("p",{parentName:"div"},"into the ggplot command to put the two plots side by side."))),(0,s.kt)("h1",{id:"investigating-a-sickle-modifier"},"Investigating a sickle modifier"),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"(",(0,s.kt)("strong",{parentName:"p"},"rs61028892")," is a single nucleotide polymorphism which has been associated with ",(0,s.kt)("a",{parentName:"p",href:"https://www.medrxiv.org/content/10.1101/2023.05.16.23289851v1.full"},"control of fetal\nhaemoglobin")," in individuals with sickle cell\ndisease.)  Can you repeat this fores tplot for the variant?  "),(0,s.kt)("p",{parentName:"div"},"You can find the ",(0,s.kt)("a",{parentName:"p",href:"https://raw.githubusercontent.com/chg-training/chg-training-resources/main/docs/statistical_modelling/introduction/data/1000_genomes_rs61028892_grouped.tsv"},"rs61028892 data\nhere"),"."),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Note.")," Unlike the O blood group example, for this SNP it probably makes sense to compute the allele frequency directly using\nthe counts from the homozygote and heterozygote individuals, something like this:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},"(\n    data\n    %>% mutate(\n        nA = 2 * rs61028892_0 + rs61028892_1,\n        nB = 2 * rs61028892_2 + rs61028892_1 + \n    )\n)\n")),(0,s.kt)("p",{parentName:"div"},"See if you can make a forest plot of this variant across populations."))))}N.isMDXComponent=!0},76427:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/forest_plot_schematic-f64e70bced0d71f19fdde5fe43b91437.png"}}]);