"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[2350],{3905:(e,t,n)=>{n.d(t,{Zo:()=>d,kt:()=>h});var i=n(67294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function s(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,i)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?s(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):s(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function r(e,t){if(null==e)return{};var n,i,a=function(e,t){if(null==e)return{};var n,i,a={},s=Object.keys(e);for(i=0;i<s.length;i++)n=s[i],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(e);for(i=0;i<s.length;i++)n=s[i],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var c=i.createContext({}),p=function(e){var t=i.useContext(c),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},d=function(e){var t=p(e.components);return i.createElement(c.Provider,{value:t},e.children)},l="mdxType",u={inlineCode:"code",wrapper:function(e){var t=e.children;return i.createElement(i.Fragment,{},t)}},f=i.forwardRef((function(e,t){var n=e.components,a=e.mdxType,s=e.originalType,c=e.parentName,d=r(e,["components","mdxType","originalType","parentName"]),l=p(n),f=a,h=l["".concat(c,".").concat(f)]||l[f]||u[f]||s;return n?i.createElement(h,o(o({ref:t},d),{},{components:n})):i.createElement(h,o({ref:t},d))}));function h(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var s=n.length,o=new Array(s);o[0]=f;var r={};for(var c in t)hasOwnProperty.call(t,c)&&(r[c]=t[c]);r.originalType=e,r[l]="string"==typeof e?e:a,o[1]=r;for(var p=2;p<s;p++)o[p]=n[p];return i.createElement.apply(null,o)}return i.createElement.apply(null,n)}f.displayName="MDXCreateElement"},13655:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>c,contentTitle:()=>o,default:()=>u,frontMatter:()=>s,metadata:()=>r,toc:()=>p});var i=n(87462),a=(n(67294),n(3905));const s={sidebar_position:11},o="Assessing effect sizes",r={unversionedId:"genome_wide_association_studies/quantitative_trait_gwas/effect_sizes",id:"genome_wide_association_studies/quantitative_trait_gwas/effect_sizes",title:"Assessing effect sizes",description:"Looking at the genomic region around our main peak, one of the genes in this region is fucosyltransferase 2 (FUT2).",source:"@site/docs/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes.md",sourceDirName:"genome_wide_association_studies/quantitative_trait_gwas",slug:"/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes.md",tags:[],version:"current",sidebarPosition:11,frontMatter:{sidebar_position:11},sidebar:"sidebar8",previous:{title:"Part 6: Interpreting GWAS signals",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/interpretation"},next:{title:"Appendix: Looking for independent signals",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/independent_signals"}},c={},p=[],d={toc:p},l="wrapper";function u(e){let{components:t,...n}=e;return(0,a.kt)(l,(0,i.Z)({},d,n,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("h1",{id:"assessing-effect-sizes"},"Assessing effect sizes"),(0,a.kt)("p",null,"Looking at the genomic region around our main peak, one of the genes in this region is fucosyltransferase 2 (FUT2).\nUsing the genetic variation data present in Ensembl, we can see that there is a known SNP (rs601338) that is associated\nwith a known phenotype (Secretor/non-secretor). The wild-type allele (G) encodes a functional copy of the FUT2 gene,\nallowing secretion of ABO histo-blood group antigens in the gastro-intestinal tract. These glycoproteins serve as a\nviral attachment factor or receptor for some strains of norovirus, influencing susceptibility to infection. A nonsense G"),(0,a.kt)("blockquote",null,(0,a.kt)("p",{parentName:"blockquote"},"A mutation at position 428 of the FUT2 gene (position 19:49206674 on GRCh37) results in a defective copy of the FUT2\ngene. Individuals homozygous for this mutation (AA; non-secretors) do not express ABO glycoproteins at the gut surface,\nproviding a protective effect against norovirus infection.")),(0,a.kt)("p",null,"We can test whether the presence of this mutation is associated with differences in antibody responses to norovirus in\nour dataset, firstly by checking if the SNP is genotyped in our data, then by testing for an effect of genotype on\nphenotype distribution. We can use this information to locate summary statistics for this SNP from our PLINK and GCTA\noutput using ",(0,a.kt)("inlineCode",{parentName:"p"},"grep"),". See the ",(0,a.kt)("a",{parentName:"p",href:"/chg-training-resources/genome_wide_association_studies/file_formats/"},"file formats")," section for what each column means. You might notice that\ngrep returns nothing for the genotyped data. This is because the SNP was not included in the original dataset, as is\noften the case when using genotype arrays. In this case, imputation has allowed us to estimate genotypes for each\nindividual."),(0,a.kt)("p",null,"Both PLINK and GCTA report betas relative to the A1 allele. An A1 frequency > 0.5 indicates that the A1 allele represents the major allele. Likewise, an A1 frequency < 0.5 indicates that the A1 allele represents the minor allele. The betas can give an indicator of the expected effect direction. In quantitative studies, a negative beta would usually be associated with the lower phenotypic value, while the positive beta would suggest that the A1 allele is associated with a higher phenotypic value.\nA simple method to test this is to extract the genotypes for each individual in the final dataset (GG/GA/AA), assign them to secretor (GG or GA) or non-secretor (AA) status, and conduct a two-sided Students T-test to test for a significant difference between secretors and non-secretors."),(0,a.kt)("p",null,"An example R script to test for differences in antibody responses to norovirus between genotypes and plot the\ndistributions as box and whisker plots is this:"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-r"},'library( ggpubr )\nlibrary( dplyr )\n\n# Read in genotypes for the prefiltered dataset\ndata <- readr::read_tsv( "Phenotype_data/AMR_geno_status.txt" )\n\n# Read in phenotypes for the filtered dataset\nphenotypes <- readr::read_tsv("Phenotype_data/AMR_phenotype.INT_transformed.txt" )\n\n# Get list of individuals retained after filtering\nkeep <- readr::read_table_("output/AMR_filtered_samples.keep.list" )\ncolnames(keep) <- c("FID", "IID")\n\n# Get genotypes for only those individuals retained in the analysis\n# ... and add the phenotypes\nfiltered_subset = (\n    data \n    %>% filter( IID %in% keep$IID )\n    %>% select( IID, Se_genotype )\n    %>% inner_join( phenotypes, by = c( "IID" ))\n)\n\n\n# Perform pairwise comparisons between genotypes and plot\nmy_comparisons <- list(\n    c("GG", "GA"),\n    c("GA", "AA"),\n    c("GG", "AA")\n)\np = (\n    ggboxplot(\n        filtered_subset,\n        x = "Se_genotype",\n        y = "Noro_response_INT"\n    )\n    + stat_compare_means(\n        comparisons = my_comparisons,\n        method="t.test",\n        paired=FALSE\n    )\n)\n\nggsave( p, "output/AMR.Norovirus_response_by_genotype.png" )\n\n')),(0,a.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,a.kt)("div",{parentName:"div",className:"admonition-heading"},(0,a.kt)("h5",{parentName:"div"},(0,a.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,a.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,a.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,a.kt)("div",{parentName:"div",className:"admonition-content"},(0,a.kt)("p",{parentName:"div"},"Amend the script to use secretor status instead of genotype to determine if these differences also occur\nwhen the population is grouped by secretor status (as described above)."))),(0,a.kt)("p",null,"For practicality, we have included the genotypes and secretor status of all individuals in the pre-filtered datasets in\n",(0,a.kt)("inlineCode",{parentName:"p"},"Phenotype_data/AMR.FUT2_geno_status.txt"),". Genotypes are encoded as ",(0,a.kt)("inlineCode",{parentName:"p"},"GG"),", ",(0,a.kt)("inlineCode",{parentName:"p"},"GA"),", ",(0,a.kt)("inlineCode",{parentName:"p"},"AA")," or ",(0,a.kt)("inlineCode",{parentName:"p"},"NA")," for those with at least one\nmissing allele. If you have time, we encourage you to try to create your own script to extract this information from the\nPLINK files yourself. "),(0,a.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,a.kt)("div",{parentName:"div",className:"admonition-heading"},(0,a.kt)("h5",{parentName:"div"},(0,a.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,a.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,a.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Questions")),(0,a.kt)("div",{parentName:"div",className:"admonition-content"},(0,a.kt)("p",{parentName:"div"},"What is the effect size for rs601338? What does this suggest about the effect of the wild-type and mutated alleles with regards to antibody responses to norovirus?"),(0,a.kt)("p",{parentName:"div"},"Are there differences in antibody response against norovirus between genotypes?"),(0,a.kt)("p",{parentName:"div"},"What about between secretors and non-secretors?"))))}u.isMDXComponent=!0}}]);