"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[9073],{3905:(e,t,a)=>{a.d(t,{Zo:()=>u,kt:()=>g});var n=a(67294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function o(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function i(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?o(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):o(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},o=Object.keys(e);for(n=0;n<o.length;n++)a=o[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(n=0;n<o.length;n++)a=o[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var s=n.createContext({}),c=function(e){var t=n.useContext(s),a=t;return e&&(a="function"==typeof e?e(t):i(i({},t),e)),a},u=function(e){var t=c(e.components);return n.createElement(s.Provider,{value:t},e.children)},p="mdxType",m={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},f=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,o=e.originalType,s=e.parentName,u=l(e,["components","mdxType","originalType","parentName"]),p=c(a),f=r,g=p["".concat(s,".").concat(f)]||p[f]||m[f]||o;return a?n.createElement(g,i(i({ref:t},u),{},{components:a})):n.createElement(g,i({ref:t},u))}));function g(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var o=a.length,i=new Array(o);i[0]=f;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l[p]="string"==typeof e?e:r,i[1]=l;for(var c=2;c<o;c++)i[c]=a[c];return n.createElement.apply(null,i)}return n.createElement.apply(null,a)}f.displayName="MDXCreateElement"},72360:(e,t,a)=>{a.d(t,{Z:()=>i});var n=a(67294),r=a(86010);const o={tabItem:"tabItem_OmH5"};function i(e){let{children:t,hidden:a,className:i}=e;return n.createElement("div",{role:"tabpanel",className:(0,r.Z)(o.tabItem,i),hidden:a},t)}},9877:(e,t,a)=>{a.d(t,{Z:()=>m});var n=a(87462),r=a(67294),o=a(72389),i=a(67392),l=a(7094),s=a(12466),c=a(86010);const u={tabList:"tabList_uSqn",tabItem:"tabItem_LplD"};function p(e){const{lazy:t,block:a,defaultValue:o,values:p,groupId:m,className:f}=e,g=r.Children.map(e.children,(e=>{if((0,r.isValidElement)(e)&&void 0!==e.props.value)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)})),d=p??g.map((e=>{let{props:{value:t,label:a,attributes:n}}=e;return{value:t,label:a,attributes:n}})),b=(0,i.l)(d,((e,t)=>e.value===t.value));if(b.length>0)throw new Error(`Docusaurus error: Duplicate values "${b.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`);const h=null===o?o:o??g.find((e=>e.props.default))?.props.value??g[0]?.props.value;if(null!==h&&!d.some((e=>e.value===h)))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${h}" but none of its children has the corresponding value. Available values are: ${d.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);const{tabGroupChoices:y,setTabGroupChoices:v}=(0,l.U)(),[k,_]=(0,r.useState)(h),w=[],{blockElementScrollPositionUntilNextRender:E}=(0,s.o5)();if(null!=m){const e=y[m];null!=e&&e!==k&&d.some((t=>t.value===e))&&_(e)}const O=e=>{const t=e.currentTarget,a=w.indexOf(t),n=d[a].value;n!==k&&(E(t),_(n),null!=m&&v(m,n))},T=e=>{let t=null;switch(e.key){case"ArrowRight":{const a=w.indexOf(e.currentTarget)+1;t=w[a]||w[0];break}case"ArrowLeft":{const a=w.indexOf(e.currentTarget)-1;t=w[a]||w[w.length-1];break}}t?.focus()};return r.createElement("div",{className:(0,c.Z)("tabs-container",u.tabList)},r.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,c.Z)("tabs",{"tabs--block":a},f)},d.map((e=>{let{value:t,label:a,attributes:o}=e;return r.createElement("li",(0,n.Z)({role:"tab",tabIndex:k===t?0:-1,"aria-selected":k===t,key:t,ref:e=>w.push(e),onKeyDown:T,onFocus:O,onClick:O},o,{className:(0,c.Z)("tabs__item",u.tabItem,o?.className,{"tabs__item--active":k===t})}),a??t)}))),t?(0,r.cloneElement)(g.filter((e=>e.props.value===k))[0],{className:"margin-top--md"}):r.createElement("div",{className:"margin-top--md"},g.map(((e,t)=>(0,r.cloneElement)(e,{key:t,hidden:e.props.value!==k})))))}function m(e){const t=(0,o.Z)();return r.createElement(p,(0,n.Z)({key:String(t)},e))}},42536:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>s,contentTitle:()=>i,default:()=>m,frontMatter:()=>o,metadata:()=>l,toc:()=>c});var n=a(87462),r=(a(67294),a(3905));a(9877),a(72360);const o={sidebar_position:5},i="Making an R or python package",l={unversionedId:"bioinformatics/making_an_R_package/README",id:"bioinformatics/making_an_R_package/README",title:"Making an R or python package",description:"Welcome!  This tutorial assumes you have worked through the Analysing gene annotation in R or python tutorial first.  If so, you have a fully-fledged function called readgff() that can read data from a GFF file.",source:"@site/docs/bioinformatics/making_an_R_package/README.md",sourceDirName:"bioinformatics/making_an_R_package",slug:"/bioinformatics/making_an_R_package/",permalink:"/chg-training-resources/bioinformatics/making_an_R_package/",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/bioinformatics/making_an_R_package/README.md",tags:[],version:"current",sidebarPosition:5,frontMatter:{sidebar_position:5},sidebar:"sidebar2",previous:{title:"Refactor for better readability",permalink:"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/appendices/refactoring"},next:{title:"The read_gff() function",permalink:"/chg-training-resources/bioinformatics/making_an_R_package/read_gff"}},s={},c=[],u={toc:c},p="wrapper";function m(e){let{components:t,...a}=e;return(0,r.kt)(p,(0,n.Z)({},u,a,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"making-an-r-or-python-package"},"Making an R or python package"),(0,r.kt)("p",null,"Welcome!  This tutorial assumes you have worked through the ",(0,r.kt)("a",{parentName:"p",href:"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/"},"Analysing gene annotation in R or python")," tutorial first.  If so, you have a fully-fledged function called ",(0,r.kt)("inlineCode",{parentName:"p"},"read_gff()")," that can read data from a ",(0,r.kt)("a",{parentName:"p",href:"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/What_gene_annotation_data_looks_like"},"GFF file"),"."),(0,r.kt)("p",null,"Let's now turn the function into an R package (or python module.)"),(0,r.kt)("p",null,"To get started, go ",(0,r.kt)("a",{parentName:"p",href:"/chg-training-resources/bioinformatics/making_an_R_package/read_gff"},"here"),"."))}m.isMDXComponent=!0}}]);