"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[7275],{3905:(e,t,n)=>{n.d(t,{Zo:()=>c,kt:()=>u});var a=n(67294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},r=Object.keys(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var p=a.createContext({}),l=function(e){var t=a.useContext(p),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},c=function(e){var t=l(e.components);return a.createElement(p.Provider,{value:t},e.children)},m="mdxType",g={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},d=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,r=e.originalType,p=e.parentName,c=s(e,["components","mdxType","originalType","parentName"]),m=l(n),d=i,u=m["".concat(p,".").concat(d)]||m[d]||g[d]||r;return n?a.createElement(u,o(o({ref:t},c),{},{components:n})):a.createElement(u,o({ref:t},c))}));function u(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=n.length,o=new Array(r);o[0]=d;var s={};for(var p in t)hasOwnProperty.call(t,p)&&(s[p]=t[p]);s.originalType=e,s[m]="string"==typeof e?e:i,o[1]=s;for(var l=2;l<r;l++)o[l]=n[l];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}d.displayName="MDXCreateElement"},4212:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>p,contentTitle:()=>o,default:()=>g,frontMatter:()=>r,metadata:()=>s,toc:()=>l});var a=n(87462),i=(n(67294),n(3905));const r={sidebar_position:1},o="Getting setup",s={unversionedId:"bioinformatics/analysing_genes_across_species/getting_setup",id:"bioinformatics/analysing_genes_across_species/getting_setup",title:"Getting setup",description:"If you followed the R version of our earlier tutorials you will have:",source:"@site/docs/bioinformatics/analysing_genes_across_species/001_getting_setup.md",sourceDirName:"bioinformatics/analysing_genes_across_species",slug:"/bioinformatics/analysing_genes_across_species/getting_setup",permalink:"/chg-training-resources/bioinformatics/analysing_genes_across_species/getting_setup",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/bioinformatics/analysing_genes_across_species/001_getting_setup.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"sidebar2",previous:{title:"Analysing genes across species",permalink:"/chg-training-resources/bioinformatics/analysing_genes_across_species/"},next:{title:"Counting genes",permalink:"/chg-training-resources/bioinformatics/analysing_genes_across_species/Counting_genes_1"}},p={},l=[{value:"A command-line program",id:"a-command-line-program",level:2},{value:"Next steps",id:"next-steps",level:2}],c={toc:l},m="wrapper";function g(e){let{components:t,...n}=e;return(0,i.kt)(m,(0,a.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"getting-setup"},"Getting setup"),(0,i.kt)("p",null,"If you followed the R version of our earlier tutorials you will have:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"An R package called ",(0,i.kt)("strong",{parentName:"li"},"gmsgff")," which provides the ",(0,i.kt)("inlineCode",{parentName:"li"},"read_gff()")," function.  (We made this in the ",(0,i.kt)("a",{parentName:"li",href:"/chg-training-resources/bioinformatics/making_an_R_package/"},"Making an R or python package")," tutorial.)"),(0,i.kt)("li",{parentName:"ul"},"An R script called ",(0,i.kt)("inlineCode",{parentName:"li"},"gff_to_sqlite.R")," which converts a GFF file into the sqlite database format.  (We made this in the ",(0,i.kt)("a",{parentName:"li",href:"/chg-training-resources/bioinformatics/a_command_line_program/"},"Making a command-line program")," tutorial.)")),(0,i.kt)("p",null,"If you don't have these or want an updated version, don't worry!  You can get my versions as follows."),(0,i.kt)("p",null,"First, to install the R package, try (from an R session):"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'install.packages(\n    "https://www.chg.ox.ac.uk/bioinformatics/training/gms/code/R/gmsgff.tgz",\n    repos = NULL,\n    type = "source"\n)\n')),(0,i.kt)("p",null,"This will install the ",(0,i.kt)("strong",{parentName:"p"},"gmsgff")," package - this is the ",(0,i.kt)("a",{parentName:"p",href:"../programming_with_gene_annotations3/007_making_a_module.md"},"same as the one you developed")," with some additions that came from the ",(0,i.kt)("a",{parentName:"p",href:"../programming_with_gene_annotations3/009_challenge_questions.md"},"challenge\nquestions"),".  "),(0,i.kt)("p",null,"For example you can use it to load GFF data like this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'data = gmsgff::read_gff( "/path/to/Homo_sapiens.GRCh38.107.chr.gff3.gz", extra_attributes = c( "biotype", "Name" ))\n')),(0,i.kt)("p",null,"You should see some nice output like this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'++read_gff(): Reading GFF3 data from "Homo_sapiens.GRCh38.107.chr.gff3.gz"...\n++read_gff(): Extracting "ID" attribute...                                                          \n++read_gff(): Extracting "Parent" attribute...\n++read_gff(): Extracting "biotype" attribute...\n++read_gff(): Extracting "Name" attribute...\n++read_gff(): Removing prefixes from ID fields...\n++read_gff(): ok.\n')),(0,i.kt)("p",null,"You should get back a dataframe with all the ",(0,i.kt)("a",{parentName:"p",href:"https://m.ensembl.org/info/website/upload/gff3.html"},"GFF columns"),", plus\nthe extracted ",(0,i.kt)("inlineCode",{parentName:"p"},"ID"),", ",(0,i.kt)("inlineCode",{parentName:"p"},"Parent"),", ",(0,i.kt)("inlineCode",{parentName:"p"},"biotype")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"Name")," attributes as seperate columns."),(0,i.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"Warning")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Remember these are big files and they use lots of memory.  It is worth having a seperate terminal running and monitoring\nthe memory usage as this data loads - use ",(0,i.kt)("inlineCode",{parentName:"p"},"top -u <username> -o '%MEM'")," (on linux) ",(0,i.kt)("inlineCode",{parentName:"p"},"top -U gav -o MEM")," (on Mac OS) to do\nthis, or use your system activity monitor.  How much memory does the process use?"))),(0,i.kt)("h2",{id:"a-command-line-program"},"A command-line program"),(0,i.kt)("p",null,"If you completed the tutorial you should also have a command-line program ",(0,i.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite.R")," which can be used to\n",(0,i.kt)("a",{parentName:"p",href:"../programming_with_gene_annotations3/008_Converting_gff_to_sqlite.md"},"convert a GFF file into sqlite format"),".  "),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"If you don't have this program, fear not!  You can download my version at this link:\n",(0,i.kt)("a",{parentName:"p",href:"https://github.com/chg-training/chg-training-resources/blob/main/docs/bioinformatics/analysing_genes_across_species/code/gff_to_sqlite.R"},"gff_to_sqlite.R"),".  Click on 'Raw', copy the code, and paste into your ",(0,i.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite.R")," file in your current directory."),(0,i.kt)("p",{parentName:"div"},"(This program depends on the ",(0,i.kt)("inlineCode",{parentName:"p"},"gmsgff")," R library above, so make sure to install that first.)"))),(0,i.kt)("p",null,"In the command-line you can now run the program like this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"Rscript --vanilla gff_to_sqlite.R --input Homo_sapiens.GRCh38.107.chr.gff3.gz --output genes.sqlite --attributes biotype Name\n")),(0,i.kt)("p",null,"which will produce a new file called ",(0,i.kt)("inlineCode",{parentName:"p"},"genes.sqlite"),".  If you're not used to using sqlite files, you can see some ways to\naccess that data ",(0,i.kt)("a",{parentName:"p",href:"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/appendices/sqlite_access"},"on this page"),"."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"We added the 'biotype' and 'Name' attributes above - these are useful in the ",(0,i.kt)("a",{parentName:"p",href:"http://ftp.ensembl.org/pub/current_gff3/"},"Ensembl files")," (but not in the Gencode files, which use ",(0,i.kt)("inlineCode",{parentName:"p"},"gene_type")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"gene_name")," instead.)"))),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Another amazing feature of this program is that it will fetch data from the internet for you!  For example, try:"),(0,i.kt)("pre",{parentName:"div"},(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"Rscript \\\n--vanilla gff_to_sqlite.R \\\n--input http://ftp.ensembl.org/pub/current_gff3/camelus_dromedarius/Camelus_dromedarius.CamDro2.110.chr.gff3.gz \\\n--attributes biotype Name \\\n--output genes.sqlite\n")),(0,i.kt)("p",{parentName:"div"},"This didn't require any new programming - it is all done by the underlying ",(0,i.kt)("inlineCode",{parentName:"p"},"read_tsv()")," function ",(0,i.kt)("a",{parentName:"p",href:"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/Getting_started_writing_some_code"},"which we used to load\ndata"),"."))),(0,i.kt)("h2",{id:"next-steps"},"Next steps"),(0,i.kt)("p",null,"You are all set to start ",(0,i.kt)("a",{parentName:"p",href:"/chg-training-resources/bioinformatics/analysing_genes_across_species/Counting_genes_1"},"counting genes"),"."))}g.isMDXComponent=!0}}]);