"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[53],{1109:i=>{i.exports=JSON.parse('{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"sidebar1":[{"type":"link","label":"Getting your environment setup","href":"/chg-training-resources/prerequisites/","docId":"prerequisites/README"},{"type":"link","label":"Setting up a UNIX terminal","href":"/chg-training-resources/prerequisites/UNIX","docId":"prerequisites/UNIX"},{"type":"link","label":"Setting up Conda","href":"/chg-training-resources/prerequisites/CONDA","docId":"prerequisites/CONDA"},{"type":"link","label":"R and RStudio","href":"/chg-training-resources/prerequisites/R","docId":"prerequisites/R"},{"type":"link","label":"Getting a text editor","href":"/chg-training-resources/prerequisites/editor","docId":"prerequisites/editor"},{"type":"link","label":"The R `tidyverse`","href":"/chg-training-resources/prerequisites/tidyverse","docId":"prerequisites/tidyverse"},{"type":"link","label":"Python","href":"/chg-training-resources/prerequisites/python","docId":"prerequisites/python"},{"type":"link","label":"Julia","href":"/chg-training-resources/prerequisites/julia","docId":"prerequisites/julia"},{"type":"link","label":"Using R, python and julia with JupyterLab","href":"/chg-training-resources/prerequisites/Jupyterlab","docId":"prerequisites/Jupyterlab"},{"type":"category","label":"Other languages","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Javascript","href":"/chg-training-resources/prerequisites/Other_languages/javascript","docId":"prerequisites/Other_languages/javascript"}],"href":"/chg-training-resources/prerequisites/Other_languages/"}],"sidebar2":[{"type":"link","label":"Bioinformatics and programming","href":"/chg-training-resources/bioinformatics/","docId":"bioinformatics/README"},{"type":"category","label":"Getting started with the command line","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Starting a terminal","href":"/chg-training-resources/bioinformatics/command_line/starting","docId":"bioinformatics/command_line/starting"},{"type":"link","label":"Command line basics","href":"/chg-training-resources/bioinformatics/command_line/basics","docId":"bioinformatics/command_line/basics"},{"type":"link","label":"Getting help!","href":"/chg-training-resources/bioinformatics/command_line/help","docId":"bioinformatics/command_line/help"},{"type":"link","label":"Standard command syntax","href":"/chg-training-resources/bioinformatics/command_line/standard_syntax","docId":"bioinformatics/command_line/standard_syntax"},{"type":"link","label":"Interacting with the filesystem","href":"/chg-training-resources/bioinformatics/command_line/tour","docId":"bioinformatics/command_line/tour"},{"type":"link","label":"Some important facts about file names","href":"/chg-training-resources/bioinformatics/command_line/important_facts_about_filenames","docId":"bioinformatics/command_line/important_facts_about_filenames"},{"type":"link","label":"Working with files","href":"/chg-training-resources/bioinformatics/command_line/working_with_files","docId":"bioinformatics/command_line/working_with_files"},{"type":"link","label":"I/O redirection and pipelining","href":"/chg-training-resources/bioinformatics/command_line/pipelines","docId":"bioinformatics/command_line/pipelines"},{"type":"link","label":"Filtering files: sort, uniq, cut, tr","href":"/chg-training-resources/bioinformatics/command_line/filtering_files","docId":"bioinformatics/command_line/filtering_files"},{"type":"category","label":"Advanced topics","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"The emergency exit","href":"/chg-training-resources/bioinformatics/command_line/appendices/emergency_brake","docId":"bioinformatics/command_line/appendices/emergency_brake"},{"type":"link","label":"Navigating history","href":"/chg-training-resources/bioinformatics/command_line/appendices/navigating_history","docId":"bioinformatics/command_line/appendices/navigating_history"},{"type":"link","label":"Wildcards and \'globbing\'","href":"/chg-training-resources/bioinformatics/command_line/appendices/globbing","docId":"bioinformatics/command_line/appendices/globbing"},{"type":"link","label":"Where are the commands?","href":"/chg-training-resources/bioinformatics/command_line/appendices/where_are_the_commands","docId":"bioinformatics/command_line/appendices/where_are_the_commands"},{"type":"link","label":"Environment variables","href":"/chg-training-resources/bioinformatics/command_line/appendices/environment_variables","docId":"bioinformatics/command_line/appendices/environment_variables"},{"type":"link","label":"Installing the Windows Subsystem for Linux","href":"/chg-training-resources/bioinformatics/command_line/appendices/installing_wsl","docId":"bioinformatics/command_line/appendices/installing_wsl"},{"type":"link","label":"A table of useful commands","href":"/chg-training-resources/bioinformatics/command_line/appendices/table_of_commands","docId":"bioinformatics/command_line/appendices/table_of_commands"}],"href":"/chg-training-resources/bioinformatics/command_line/appendices/"},{"type":"link","label":"Challenge questions","href":"/chg-training-resources/bioinformatics/command_line/questions","docId":"bioinformatics/command_line/questions"}],"href":"/chg-training-resources/bioinformatics/command_line/"},{"type":"category","label":"Introduction to R","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"R as a calculator","href":"/chg-training-resources/bioinformatics/introduction_to_R/fundamentals","docId":"bioinformatics/introduction_to_R/fundamentals"},{"type":"link","label":"Computing with lots of values at once","href":"/chg-training-resources/bioinformatics/introduction_to_R/vectors","docId":"bioinformatics/introduction_to_R/vectors"},{"type":"link","label":"Help! it\'s not working!","href":"/chg-training-resources/bioinformatics/introduction_to_R/help_going_wrong","docId":"bioinformatics/introduction_to_R/help_going_wrong"},{"type":"link","label":"Storing values in variables","href":"/chg-training-resources/bioinformatics/introduction_to_R/variables","docId":"bioinformatics/introduction_to_R/variables"},{"type":"link","label":"Some other data types","href":"/chg-training-resources/bioinformatics/introduction_to_R/data_types","docId":"bioinformatics/introduction_to_R/data_types"},{"type":"link","label":"Functions","href":"/chg-training-resources/bioinformatics/introduction_to_R/functions","docId":"bioinformatics/introduction_to_R/functions"},{"type":"link","label":"Getting around","href":"/chg-training-resources/bioinformatics/introduction_to_R/getting_around","docId":"bioinformatics/introduction_to_R/getting_around"},{"type":"link","label":"Loading libraries","href":"/chg-training-resources/bioinformatics/introduction_to_R/libaries","docId":"bioinformatics/introduction_to_R/libaries"},{"type":"link","label":"Working with data","href":"/chg-training-resources/bioinformatics/introduction_to_R/working_with_data","docId":"bioinformatics/introduction_to_R/working_with_data"},{"type":"link","label":"Summary","href":"/chg-training-resources/bioinformatics/introduction_to_R/summary","docId":"bioinformatics/introduction_to_R/summary"}],"href":"/chg-training-resources/bioinformatics/introduction_to_R/"},{"type":"category","label":"Exploring gene annotations on the command-line","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting started","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/getting_started","docId":"bioinformatics/exploring_gene_annotations_in_bash/getting_started"},{"type":"link","label":"Decompressing the file","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/decompressing","docId":"bioinformatics/exploring_gene_annotations_in_bash/decompressing"},{"type":"link","label":"Viewing gene records","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/viewing","docId":"bioinformatics/exploring_gene_annotations_in_bash/viewing"},{"type":"link","label":"Counting genes","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/counting","docId":"bioinformatics/exploring_gene_annotations_in_bash/counting"},{"type":"link","label":"Counting genes again","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/counting2","docId":"bioinformatics/exploring_gene_annotations_in_bash/counting2"},{"type":"link","label":"Investigating specific genes","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/investigating","docId":"bioinformatics/exploring_gene_annotations_in_bash/investigating"},{"type":"link","label":"Browsing the genome","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/browsing","docId":"bioinformatics/exploring_gene_annotations_in_bash/browsing"},{"type":"link","label":"Challenge questions","href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/questions","docId":"bioinformatics/exploring_gene_annotations_in_bash/questions"}],"href":"/chg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/"},{"type":"category","label":"Analysing gene annotations in R or python","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Introduction","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/Introduction","docId":"bioinformatics/programming_with_gene_annotations3/Introduction"},{"type":"link","label":"What gene annotation data looks like","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/What_gene_annotation_data_looks_like","docId":"bioinformatics/programming_with_gene_annotations3/What_gene_annotation_data_looks_like"},{"type":"link","label":"A first go at parsing GFF","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/Getting_started_writing_some_code","docId":"bioinformatics/programming_with_gene_annotations3/Getting_started_writing_some_code"},{"type":"link","label":"Extracting the ID and Parent attributes","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/extracting_attributes","docId":"bioinformatics/programming_with_gene_annotations3/extracting_attributes"},{"type":"link","label":"Extracting more attributes","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/004b_more_attributes","docId":"bioinformatics/programming_with_gene_annotations3/004b_more_attributes"},{"type":"link","label":"Testing it out","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/testing_it_out","docId":"bioinformatics/programming_with_gene_annotations3/testing_it_out"},{"type":"link","label":"Quick recap","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/recap","docId":"bioinformatics/programming_with_gene_annotations3/recap"},{"type":"category","label":"appendices","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Reading results from a sqlite file in any language","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/appendices/sqlite_access","docId":"bioinformatics/programming_with_gene_annotations3/appendices/sqlite_access"},{"type":"link","label":"Refactor for better readability","href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/appendices/refactoring","docId":"bioinformatics/programming_with_gene_annotations3/appendices/refactoring"}]}],"href":"/chg-training-resources/bioinformatics/programming_with_gene_annotations3/"},{"type":"category","label":"Making an R or python package","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"The read_gff() function","href":"/chg-training-resources/bioinformatics/making_an_R_package/read_gff","docId":"bioinformatics/making_an_R_package/read_gff"},{"type":"link","label":"Taking the easy way","href":"/chg-training-resources/bioinformatics/making_an_R_package/single_file","docId":"bioinformatics/making_an_R_package/single_file"},{"type":"link","label":"A fully-fledged R package","href":"/chg-training-resources/bioinformatics/making_an_R_package/fully_fledged","docId":"bioinformatics/making_an_R_package/fully_fledged"}],"href":"/chg-training-resources/bioinformatics/making_an_R_package/"},{"type":"category","label":"Writing a command-line program","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"A command-line program","href":"/chg-training-resources/bioinformatics/a_command_line_program/a_program","docId":"bioinformatics/a_command_line_program/a_program"},{"type":"link","label":"Challenge questions","href":"/chg-training-resources/bioinformatics/a_command_line_program/challenge_questions","docId":"bioinformatics/a_command_line_program/challenge_questions"}],"href":"/chg-training-resources/bioinformatics/a_command_line_program/"},{"type":"category","label":"Analysing genes across species","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting setup","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/getting_setup","docId":"bioinformatics/analysing_genes_across_species/getting_setup"},{"type":"link","label":"Counting genes","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/Counting_genes_1","docId":"bioinformatics/analysing_genes_across_species/Counting_genes_1"},{"type":"link","label":"What\'s not a gene?","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/Counting_genes_2","docId":"bioinformatics/analysing_genes_across_species/Counting_genes_2"},{"type":"link","label":"Filter, group, summarise, join, arrange...","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/filter_join_merge","docId":"bioinformatics/analysing_genes_across_species/filter_join_merge"},{"type":"category","label":"Extreme genes!","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Gene length","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_1","docId":"bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_1"},{"type":"link","label":"Computing coding length","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_2","docId":"bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_2"},{"type":"link","label":"Picking a \'canonical\' transcript","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/canonical_transcripts","docId":"bioinformatics/analysing_genes_across_species/extreme_genes/canonical_transcripts"},{"type":"link","label":"The number of exons","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/how_many_exons","docId":"bioinformatics/analysing_genes_across_species/extreme_genes/how_many_exons"},{"type":"link","label":"Extreme genes revisited","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/extreme_genes_revisited","docId":"bioinformatics/analysing_genes_across_species/extreme_genes/extreme_genes_revisited"},{"type":"link","label":"Too long, didn\'t read","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/tldr","docId":"bioinformatics/analysing_genes_across_species/extreme_genes/tldr"}],"href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/"},{"type":"link","label":"Getting chromosome lengths","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/getting_chromosomes","docId":"bioinformatics/analysing_genes_across_species/getting_chromosomes"},{"type":"category","label":"How much of the genome is in genes?","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"A naive approach","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/naive_approach","docId":"bioinformatics/analysing_genes_across_species/how_much_in_genes/naive_approach"},{"type":"link","label":"An algorithm to merge regions","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/an_algorithm_to_merge_regions","docId":"bioinformatics/analysing_genes_across_species/how_much_in_genes/an_algorithm_to_merge_regions"},{"type":"link","label":"How much of the genome is *really* in genes?","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_II","docId":"bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_II"},{"type":"link","label":"How much of the genome is in exons?","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_in_exons","docId":"bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_in_exons"}],"href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/"},{"type":"link","label":"Conclusions and challenges","href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/challenges","docId":"bioinformatics/analysing_genes_across_species/challenges"}],"href":"/chg-training-resources/bioinformatics/analysing_genes_across_species/"},{"type":"category","label":"Tips and tricks","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"List contents of an FTP site","href":"/chg-training-resources/bioinformatics/tips_and_tricks/recursive_ftp","docId":"bioinformatics/tips_and_tricks/recursive_ftp"}],"href":"/chg-training-resources/bioinformatics/tips_and_tricks/"}],"sidebar3":[{"type":"link","label":"Data visualisation tutorials","href":"/chg-training-resources/data_visualisation/","docId":"data_visualisation/README"},{"type":"category","label":"Visualise RNA-seq data in R","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting setup","href":"/chg-training-resources/data_visualisation/visualising_rna-seq_data/getting_setup","docId":"data_visualisation/visualising_rna-seq_data/getting_setup"},{"type":"link","label":"Visualising the data","href":"/chg-training-resources/data_visualisation/visualising_rna-seq_data/visualising","docId":"data_visualisation/visualising_rna-seq_data/visualising"},{"type":"category","label":"appendices","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting information about your R session","href":"/chg-training-resources/data_visualisation/visualising_rna-seq_data/appendices/getting_session_info","docId":"data_visualisation/visualising_rna-seq_data/appendices/getting_session_info"}]}],"href":"/chg-training-resources/data_visualisation/visualising_rna-seq_data/"},{"type":"category","label":"More data visualisation in R","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"A first look at Gapminder data","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/gapminder","docId":"data_visualisation/data_visualisation_with_ggplot2/gapminder"},{"type":"link","label":"The distribution of GDP per capita","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/distribution","docId":"data_visualisation/data_visualisation_with_ggplot2/distribution"},{"type":"link","label":"The distribution of GDP per capita, by continent","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/distribution2","docId":"data_visualisation/data_visualisation_with_ggplot2/distribution2"},{"type":"link","label":"Bring on the violins","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/violins","docId":"data_visualisation/data_visualisation_with_ggplot2/violins"},{"type":"link","label":"Adding summary statistics","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/summary_statistics","docId":"data_visualisation/data_visualisation_with_ggplot2/summary_statistics"},{"type":"link","label":"Reordering values","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/reordering","docId":"data_visualisation/data_visualisation_with_ggplot2/reordering"},{"type":"link","label":"Is gdp related to life expectancy?","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/gdp_life_expectancy","docId":"data_visualisation/data_visualisation_with_ggplot2/gdp_life_expectancy"},{"type":"link","label":"Labelling points","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/labelling","docId":"data_visualisation/data_visualisation_with_ggplot2/labelling"},{"type":"link","label":"Saving your figures","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/saving","docId":"data_visualisation/data_visualisation_with_ggplot2/saving"},{"type":"link","label":"Facetting","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/facetting","docId":"data_visualisation/data_visualisation_with_ggplot2/facetting"},{"type":"link","label":"Other plots - and a challenge","href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/other_plots","docId":"data_visualisation/data_visualisation_with_ggplot2/other_plots"}],"href":"/chg-training-resources/data_visualisation/data_visualisation_with_ggplot2/"}],"sidebar4":[{"type":"category","label":"Introduction to snakemake","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting set up","href":"/chg-training-resources/programming/pipelining_with_snakemake/prerequisites","docId":"programming/pipelining_with_snakemake/prerequisites"},{"type":"link","label":"Using snakemake","href":"/chg-training-resources/programming/pipelining_with_snakemake/using_snakemake","docId":"programming/pipelining_with_snakemake/using_snakemake"},{"type":"link","label":"snakefile3","href":"/chg-training-resources/programming/pipelining_with_snakemake/snakefile3","docId":"programming/pipelining_with_snakemake/snakefile3"}],"href":"/chg-training-resources/programming/pipelining_with_snakemake/"}],"sidebar5":[{"type":"category","label":"Introduction to snakemake","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting set up","href":"/chg-training-resources/programming/pipelining_with_snakemake/prerequisites","docId":"programming/pipelining_with_snakemake/prerequisites"},{"type":"link","label":"Using snakemake","href":"/chg-training-resources/programming/pipelining_with_snakemake/using_snakemake","docId":"programming/pipelining_with_snakemake/using_snakemake"},{"type":"link","label":"snakefile3","href":"/chg-training-resources/programming/pipelining_with_snakemake/snakefile3","docId":"programming/pipelining_with_snakemake/snakefile3"}],"href":"/chg-training-resources/programming/pipelining_with_snakemake/"}],"sidebar6":[{"type":"link","label":"Statistical modelling","href":"/chg-training-resources/statistical_modelling/","docId":"statistical_modelling/README"},{"type":"category","label":"From basics to Bayes","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Where statistics fits in","href":"/chg-training-resources/statistical_modelling/introduction/diagram","docId":"statistical_modelling/introduction/diagram"},{"type":"link","label":"Statistics is reasoning with uncertainty","href":"/chg-training-resources/statistical_modelling/introduction/reasoning","docId":"statistical_modelling/introduction/reasoning"},{"type":"link","label":"Distributions","href":"/chg-training-resources/statistical_modelling/introduction/distributions","docId":"statistical_modelling/introduction/distributions"},{"type":"link","label":"Distribution summaries","href":"/chg-training-resources/statistical_modelling/introduction/summarising_distributions","docId":"statistical_modelling/introduction/summarising_distributions"},{"type":"link","label":"Some useful probability distributions","href":"/chg-training-resources/statistical_modelling/introduction/some_distributions","docId":"statistical_modelling/introduction/some_distributions"},{"type":"link","label":"Quantiles, p-values and all that","href":"/chg-training-resources/statistical_modelling/introduction/quantiles_and_pvalues","docId":"statistical_modelling/introduction/quantiles_and_pvalues"},{"type":"link","label":"Toy example 1: a drug trial","href":"/chg-training-resources/statistical_modelling/introduction/drug_example","docId":"statistical_modelling/introduction/drug_example"},{"type":"link","label":"The most important formula in science","href":"/chg-training-resources/statistical_modelling/introduction/bayes","docId":"statistical_modelling/introduction/bayes"},{"type":"link","label":"What to make of a COVID test","href":"/chg-training-resources/statistical_modelling/introduction/covid","docId":"statistical_modelling/introduction/covid"},{"type":"link","label":"The bayesian update","href":"/chg-training-resources/statistical_modelling/introduction/bayesian_update","docId":"statistical_modelling/introduction/bayesian_update"},{"type":"link","label":"Estimating allele frequencies","href":"/chg-training-resources/statistical_modelling/introduction/allele_frequencies","docId":"statistical_modelling/introduction/allele_frequencies"},{"type":"link","label":"A bayesian forest plot","href":"/chg-training-resources/statistical_modelling/introduction/allele_frequencies_2","docId":"statistical_modelling/introduction/allele_frequencies_2"},{"type":"link","label":"The normal approximation","href":"/chg-training-resources/statistical_modelling/introduction/challenge","docId":"statistical_modelling/introduction/challenge"}],"href":"/chg-training-resources/statistical_modelling/introduction/"},{"type":"link","label":"Probability cheatsheet","href":"/chg-training-resources/statistical_modelling/probability_cheatsheet","docId":"statistical_modelling/probability_cheatsheet"},{"type":"link","label":"Distributions cheatsheet","href":"/chg-training-resources/statistical_modelling/distributions_cheatsheet","docId":"statistical_modelling/distributions_cheatsheet"},{"type":"category","label":"Modelling statistical association with regression","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"A linear regression example","href":"/chg-training-resources/statistical_modelling/regression_modelling/linear_regression_1","docId":"statistical_modelling/regression_modelling/linear_regression_1"},{"type":"link","label":"Linear regression cheatsheet","href":"/chg-training-resources/statistical_modelling/regression_modelling/linear_regression_cheatsheet","docId":"statistical_modelling/regression_modelling/linear_regression_cheatsheet"},{"type":"link","label":"A logistic regression example","href":"/chg-training-resources/statistical_modelling/regression_modelling/logistic_regression_1","docId":"statistical_modelling/regression_modelling/logistic_regression_1"},{"type":"link","label":"How to interpret the P-value","href":"/chg-training-resources/statistical_modelling/regression_modelling/interpreting_p_values","docId":"statistical_modelling/regression_modelling/interpreting_p_values"},{"type":"link","label":"Causal thinking","href":"/chg-training-resources/statistical_modelling/regression_modelling/causal_graphs","docId":"statistical_modelling/regression_modelling/causal_graphs"}],"href":"/chg-training-resources/statistical_modelling/regression_modelling/"},{"type":"category","label":"Glycophorin CNV calling tutorial","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Warmup: examining read coverage in a region","href":"/chg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup","docId":"statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup"},{"type":"link","label":"Maximum likelihood copy number inference","href":"/chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation","docId":"statistical_modelling/Hidden_markov_models/modelling_copy_number_variation"},{"type":"link","label":"Bayesian copy number inference","href":"/chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation_bayes","docId":"statistical_modelling/Hidden_markov_models/modelling_copy_number_variation_bayes"},{"type":"link","label":"Modelling CNVs using a HMM","href":"/chg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm","docId":"statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm"}],"href":"/chg-training-resources/statistical_modelling/Hidden_markov_models/"},{"type":"category","label":"notes","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Asymptotics and  P-values","href":"/chg-training-resources/statistical_modelling/notes/computing_pvalues","docId":"statistical_modelling/notes/computing_pvalues"}]}],"sidebar7":[{"type":"link","label":"Population genetics","href":"/chg-training-resources/population_genetics/","docId":"population_genetics/README"},{"type":"category","label":"Visualising haplotypes in R","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"tbc","href":"/chg-training-resources/population_genetics/plotting_haplotypes/tbc","docId":"population_genetics/plotting_haplotypes/tbc"}],"href":"/chg-training-resources/population_genetics/plotting_haplotypes/"},{"type":"category","label":"Simulating populations","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Genetic drift","href":"/chg-training-resources/population_genetics/simulation/popgen_simulation","docId":"population_genetics/simulation/popgen_simulation"},{"type":"link","label":"Adding mutations","href":"/chg-training-resources/population_genetics/simulation/mutations","docId":"population_genetics/simulation/mutations"},{"type":"link","label":"Some population-genetic statistics","href":"/chg-training-resources/population_genetics/simulation/statistics","docId":"population_genetics/simulation/statistics"}],"href":"/chg-training-resources/population_genetics/simulation/"},{"type":"category","label":"Principal components analysis","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"A warm-up example","href":"/chg-training-resources/population_genetics/principal_components_analysis/simulated_pca_example","docId":"population_genetics/principal_components_analysis/simulated_pca_example"},{"type":"link","label":"Getting set up","href":"/chg-training-resources/population_genetics/principal_components_analysis/getting_setup","docId":"population_genetics/principal_components_analysis/getting_setup"},{"type":"link","label":"Overview of the practical","href":"/chg-training-resources/population_genetics/principal_components_analysis/overview","docId":"population_genetics/principal_components_analysis/overview"},{"type":"link","label":"Removing highly-correlated SNPs","href":"/chg-training-resources/population_genetics/principal_components_analysis/ld_pruning","docId":"population_genetics/principal_components_analysis/ld_pruning"},{"type":"link","label":"Removing closely-related samples","href":"/chg-training-resources/population_genetics/principal_components_analysis/relatedness_pruning","docId":"population_genetics/principal_components_analysis/relatedness_pruning"},{"type":"link","label":"Computing principal components","href":"/chg-training-resources/population_genetics/principal_components_analysis/computing_PCs","docId":"population_genetics/principal_components_analysis/computing_PCs"},{"type":"link","label":"Plotting samples against a global reference panel","href":"/chg-training-resources/population_genetics/principal_components_analysis/global_analysis","docId":"population_genetics/principal_components_analysis/global_analysis"},{"type":"link","label":"An aside on the maths","href":"/chg-training-resources/population_genetics/principal_components_analysis/the_maths","docId":"population_genetics/principal_components_analysis/the_maths"}],"href":"/chg-training-resources/population_genetics/principal_components_analysis/"},{"type":"category","label":"Population structure and admixture","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting setup","href":"/chg-training-resources/population_genetics/admixture_and_biobanks/getting_setup","docId":"population_genetics/admixture_and_biobanks/getting_setup"},{"type":"link","label":"Fst","href":"/chg-training-resources/population_genetics/admixture_and_biobanks/fst","docId":"population_genetics/admixture_and_biobanks/fst"},{"type":"link","label":"Principal component analysis of the 1000 Genomes Project","href":"/chg-training-resources/population_genetics/admixture_and_biobanks/pca","docId":"population_genetics/admixture_and_biobanks/pca"},{"type":"link","label":"ADMIXTURE analysis","href":"/chg-training-resources/population_genetics/admixture_and_biobanks/admixture","docId":"population_genetics/admixture_and_biobanks/admixture"}],"href":"/chg-training-resources/population_genetics/admixture_and_biobanks/"}],"sidebar8":[{"type":"link","label":"Genome-wide association analysis","href":"/chg-training-resources/genome_wide_association_studies/","docId":"genome_wide_association_studies/README"},{"type":"category","label":"Genome-wide association study example: a case-control trait","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting setup","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/getting_setup","docId":"genome_wide_association_studies/genome_wide_association_analysis/getting_setup"},{"type":"link","label":"Plink warm-up","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies","docId":"genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies"},{"type":"link","label":"A toy GWAS","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink","docId":"genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink"},{"type":"link","label":"A real (simulated) GWAS: quality control","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control","docId":"genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control"},{"type":"link","label":"A real (simulated) GWAS: testing for association","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_association","docId":"genome_wide_association_studies/genome_wide_association_analysis/gwas_association"},{"type":"link","label":"Controlling for population structure","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/population_structure","docId":"genome_wide_association_studies/genome_wide_association_analysis/population_structure"},{"type":"link","label":"Making a regional association plot","href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/hitplots","docId":"genome_wide_association_studies/genome_wide_association_analysis/hitplots"}],"href":"/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/"},{"type":"category","label":"Genome-wide association with quantitative antibody levels","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting the data","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data","docId":"genome_wide_association_studies/quantitative_trait_gwas/getting_the_data"},{"type":"link","label":"Getting the software","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software","docId":"genome_wide_association_studies/quantitative_trait_gwas/getting_the_software"},{"type":"link","label":"Part 1: Quality control of genotype data","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes","docId":"genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes"},{"type":"link","label":"Estimating relatedness","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/relatedness","docId":"genome_wide_association_studies/quantitative_trait_gwas/relatedness"},{"type":"link","label":"Aside: excluding related samples","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/sample_relatedness","docId":"genome_wide_association_studies/quantitative_trait_gwas/sample_relatedness"},{"type":"link","label":"Part 2: Transforming the phenotype data","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/phenotype_qc","docId":"genome_wide_association_studies/quantitative_trait_gwas/phenotype_qc"},{"type":"link","label":"Part 3: Running a simple regression analysis","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/testing_for_association","docId":"genome_wide_association_studies/quantitative_trait_gwas/testing_for_association"},{"type":"link","label":"Part 4: Working with population substructure and multi-ethnic populations","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/population_structure","docId":"genome_wide_association_studies/quantitative_trait_gwas/population_structure"},{"type":"link","label":"Part 5: Imputation","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/imputed_data","docId":"genome_wide_association_studies/quantitative_trait_gwas/imputed_data"},{"type":"link","label":"Part 6: Interpreting GWAS signals","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/interpretation","docId":"genome_wide_association_studies/quantitative_trait_gwas/interpretation"},{"type":"link","label":"Assessing effect sizes","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes","docId":"genome_wide_association_studies/quantitative_trait_gwas/effect_sizes"},{"type":"link","label":"Appendix: Looking for independent signals","href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/independent_signals","docId":"genome_wide_association_studies/quantitative_trait_gwas/independent_signals"}],"href":"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/"},{"type":"category","label":"Meta-analysis and fine-mapping tutorial","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Meta-analysis and fine-mapping practical","href":"/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction","docId":"genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction"},{"type":"link","label":"Meta-analysing two studies.","href":"/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis","docId":"genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis"},{"type":"link","label":"Using FINEMAP to fine-map associations","href":"/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping","docId":"genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping"}],"href":"/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/"},{"type":"category","label":"GWAS file formats","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"PLINK file formats","href":"/chg-training-resources/genome_wide_association_studies/file_formats/plink","docId":"genome_wide_association_studies/file_formats/plink"},{"type":"link","label":"GCTA file formats","href":"/chg-training-resources/genome_wide_association_studies/file_formats/gcta","docId":"genome_wide_association_studies/file_formats/gcta"},{"type":"link","label":"\'Oxford\' file formats format","href":"/chg-training-resources/genome_wide_association_studies/file_formats/oxford","docId":"genome_wide_association_studies/file_formats/oxford"}],"href":"/chg-training-resources/genome_wide_association_studies/file_formats/"}],"sidebar9":[{"type":"link","label":"Next-generation sequencing","href":"/chg-training-resources/sequence_data_analysis/","docId":"sequence_data_analysis/README"},{"type":"category","label":"Analysing Sanger sequence data","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Viewing Sanger trace files","href":"/chg-training-resources/sequence_data_analysis/sanger_sequence_data/introduction","docId":"sequence_data_analysis/sanger_sequence_data/introduction"}],"href":"/chg-training-resources/sequence_data_analysis/sanger_sequence_data/"},{"type":"category","label":"Analysing paired-end sequencing data","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting the software","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites"},{"type":"link","label":"Getting the data","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/getting_the_data","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/getting_the_data"},{"type":"link","label":"Practical outline","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline"},{"type":"link","label":"Step 1: What\'s in the fastq files?","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs"},{"type":"link","label":"Step 2: Inspecting sequence data quality","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control"},{"type":"link","label":"Step 3: Aligning reads","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads"},{"type":"link","label":"Step 4: Viewing alignments","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments"},{"type":"link","label":"Challenge questions","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions"},{"type":"link","label":"Bonus tutorial: how accurate is Illumina sequencing?","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation"},{"type":"link","label":"Appendix 1: Paired-end sequencing in theory and practice","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory"},{"type":"link","label":"Appendix 2: understanding sequence duplication levels","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates"},{"type":"link","label":"Appendix 3: read trimming","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming"},{"type":"link","label":"Appendix 4: RNA-seq links","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq"},{"type":"link","label":"Useful links","href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links","docId":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links"}],"href":"/chg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"},{"type":"category","label":"Exploring Reads in IGV","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Looking at SNPs","href":"/chg-training-resources/sequence_data_analysis/IGV/SNPs","docId":"sequence_data_analysis/IGV/SNPs"},{"type":"link","label":"Looking at insertions and deletions","href":"/chg-training-resources/sequence_data_analysis/IGV/indels","docId":"sequence_data_analysis/IGV/indels"},{"type":"link","label":"Viewing reads - the basics","href":"/chg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV","docId":"sequence_data_analysis/IGV/Exploring_reads_in_IGV"},{"type":"link","label":"Hard-to-align regions","href":"/chg-training-resources/sequence_data_analysis/IGV/hard_to_map","docId":"sequence_data_analysis/IGV/hard_to_map"},{"type":"link","label":"Examining larger structural variants","href":"/chg-training-resources/sequence_data_analysis/IGV/SVs","docId":"sequence_data_analysis/IGV/SVs"}],"href":"/chg-training-resources/sequence_data_analysis/IGV/"},{"type":"category","label":"A sequence data pipelining challenge!","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Introduction","href":"/chg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/Introduction","docId":"sequence_data_analysis/building_an_ngs_pipeline/Introduction"},{"type":"link","label":"The pipeline","href":"/chg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/pipeline","docId":"sequence_data_analysis/building_an_ngs_pipeline/pipeline"},{"type":"link","label":"Tips and tricks","href":"/chg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/tips_and_tricks","docId":"sequence_data_analysis/building_an_ngs_pipeline/tips_and_tricks"}],"href":"/chg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/"},{"type":"category","label":"Variant calling, phasing and imputation","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Getting the data","href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Prerequisites","docId":"sequence_data_analysis/variant_calling_and_imputation/Prerequisites"},{"type":"link","label":"Variant calling with bcftools","href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Variant_calling","docId":"sequence_data_analysis/variant_calling_and_imputation/Variant_calling"},{"type":"link","label":"Variant quality control","href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Variant_quality_control","docId":"sequence_data_analysis/variant_calling_and_imputation/Variant_quality_control"},{"type":"link","label":"Phasing the variant calls","href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Phasing","docId":"sequence_data_analysis/variant_calling_and_imputation/Phasing"},{"type":"link","label":"Imputing a set of microarray data","href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Imputation","docId":"sequence_data_analysis/variant_calling_and_imputation/Imputation"},{"type":"link","label":"Challenge questions on imputation","href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Challenge_questions","docId":"sequence_data_analysis/variant_calling_and_imputation/Challenge_questions"}],"href":"/chg-training-resources/sequence_data_analysis/variant_calling_and_imputation/"},{"type":"category","label":"IGV sightseeing tour","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Dataset A: Illumina paired-end","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_a","docId":"sequence_data_analysis/sightseeing_tour/dataset_a"},{"type":"link","label":"Datasets B and C: long reads","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_b","docId":"sequence_data_analysis/sightseeing_tour/dataset_b"},{"type":"link","label":"DNA modifications","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_c","docId":"sequence_data_analysis/sightseeing_tour/dataset_c"},{"type":"link","label":"Dataset D: Illumina RNA-seq","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_d","docId":"sequence_data_analysis/sightseeing_tour/dataset_d"},{"type":"link","label":"Dataset E: ATAC-seq","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_e","docId":"sequence_data_analysis/sightseeing_tour/dataset_e"},{"type":"link","label":"Dataset E: 10X linked-read sequencing","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_f","docId":"sequence_data_analysis/sightseeing_tour/dataset_f"},{"type":"link","label":"New! Dataset G: long-read RNA-seq","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/dataset_g","docId":"sequence_data_analysis/sightseeing_tour/dataset_g"},{"type":"link","label":"Play \'guess the dataset\'","href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/quiz","docId":"sequence_data_analysis/sightseeing_tour/quiz"}],"href":"/chg-training-resources/sequence_data_analysis/sightseeing_tour/"}],"sidebar10":[{"type":"link","label":"RNA-seq and other omics","href":"/chg-training-resources/rna_seq_and_other_omics/","docId":"rna_seq_and_other_omics/README"},{"type":"category","label":"RNA-Seq data analysis","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Data preprocessing","href":"/chg-training-resources/rna_seq_and_other_omics/intro_to_rna_seq_analysis/Data_preprocessing","docId":"rna_seq_and_other_omics/intro_to_rna_seq_analysis/Data_preprocessing"},{"type":"link","label":"Identifying differentially expressed genes","href":"/chg-training-resources/rna_seq_and_other_omics/intro_to_rna_seq_analysis/Diff_exp_genes","docId":"rna_seq_and_other_omics/intro_to_rna_seq_analysis/Diff_exp_genes"}],"href":"/chg-training-resources/rna_seq_and_other_omics/intro_to_rna_seq_analysis/"}]},"docs":{"bioinformatics/a_command_line_program/a_program":{"id":"bioinformatics/a_command_line_program/a_program","title":"A command-line program","description":"We\'ll write a program called gfftosqlite.R (for the R version) or gfftosqlite.py (for the python version).","sidebar":"sidebar2"},"bioinformatics/a_command_line_program/challenge_questions":{"id":"bioinformatics/a_command_line_program/challenge_questions","title":"Challenge questions","description":"The readgff() function has an extra argument, extraattributes, that allows it to extract more attributes than just \'ID\' and \'Parent\'.","sidebar":"sidebar2"},"bioinformatics/a_command_line_program/README":{"id":"bioinformatics/a_command_line_program/README","title":"Writing a command-line program","description":"If you\'ve followed the tutorials so far, you\'ll have an R or python \'gmsgff\' package that can load data from GFF files (like those from Gencode).","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/challenges":{"id":"bioinformatics/analysing_genes_across_species/challenges","title":"Conclusions and challenges","description":"If you\'ve made it through this tutorial and the earlier part -","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/Counting_genes_1":{"id":"bioinformatics/analysing_genes_across_species/Counting_genes_1","title":"Counting genes","description":"For simplicity, from here on in we will work with the GFF3 files from the [Ensembl ftp","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/Counting_genes_2":{"id":"bioinformatics/analysing_genes_across_species/Counting_genes_2","title":"What\'s not a gene?","description":"Above we counted all \\"genes\\".  But what exactly is a \\"gene\\"?","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/canonical_transcripts":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/canonical_transcripts","title":"Picking a \'canonical\' transcript","description":"To assign a \'coding length\' to each gene, we have to deal with the fact that genes have multiple transcripts.  Suggestion: let\'s solve this by picking one \'represenative\' transcript per gene - the one with the longest coding length.","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/extreme_genes_revisited":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/extreme_genes_revisited","title":"Extreme genes revisited","description":"Phew!  If you reached this point you have computed the coding sequence length of each transcript, and have produced a table of genes linked to a \'canonical\' transcript.  You may also have computed the number of exons in each gene.","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/how_many_exons":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/how_many_exons","title":"The number of exons","description":"If you follow steps similar to the above, you ought to know how you could also add columns to the dataset reflecting the","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_1":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_1","title":"Gene length","description":"To find the longest genes as measured by length on the genome, a good place to start would be to histogram gene","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_2":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/long_genes_2","title":"Computing coding length","description":"If you followed so far you\'ll have looked at some very long genes - some of them are pretty interesting.","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/README":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/README","title":"Extreme genes!","description":"We just saw an example of an extreme gene - a gene with more","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/extreme_genes/tldr":{"id":"bioinformatics/analysing_genes_across_species/extreme_genes/tldr","title":"Too long, didn\'t read","description":"Here is all the code from this section in one place for reference.  I\'ve slightly reorganised in places.","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/filter_join_merge":{"id":"bioinformatics/analysing_genes_across_species/filter_join_merge","title":"Filter, group, summarise, join, arrange...","description":"Now is a good point to introduce in a bit more detail a set of data manipulation verbs that make working with data","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/getting_chromosomes":{"id":"bioinformatics/analysing_genes_across_species/getting_chromosomes","title":"Getting chromosome lengths","description":"To answer questions like \'how much of the genome is in genes\' we need to know what the length of each genome is.","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/getting_setup":{"id":"bioinformatics/analysing_genes_across_species/getting_setup","title":"Getting setup","description":"If you followed the R version of our earlier tutorials you will have:","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/how_much_in_genes/an_algorithm_to_merge_regions":{"id":"bioinformatics/analysing_genes_across_species/how_much_in_genes/an_algorithm_to_merge_regions","title":"An algorithm to merge regions","description":"What was wrong?","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_II":{"id":"bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_II","title":"How much of the genome is *really* in genes?","description":"If you followed the previous section you\'ll have a function mergeregions()","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_in_exons":{"id":"bioinformatics/analysing_genes_across_species/how_much_in_genes/how_much_in_exons","title":"How much of the genome is in exons?","description":"You might be a bit underwhelmed by the difference between our \'naive\' and \'correct\' calculations for genes. However the","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/how_much_in_genes/naive_approach":{"id":"bioinformatics/analysing_genes_across_species/how_much_in_genes/naive_approach","title":"A naive approach","description":"Figuring out the proportion in genes should be pretty easy using the dplyr data verbs,","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/how_much_in_genes/README":{"id":"bioinformatics/analysing_genes_across_species/how_much_in_genes/README","title":"How much of the genome is in genes?","description":"If you\'ve followed so far you should have:","sidebar":"sidebar2"},"bioinformatics/analysing_genes_across_species/README":{"id":"bioinformatics/analysing_genes_across_species/README","title":"Analysing genes across species","description":"Welcome!","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/emergency_brake":{"id":"bioinformatics/command_line/appendices/emergency_brake","title":"The emergency exit","description":"If you get stuck - for example if you don\'t seem to see a command prompt, or can\'t seem to run a","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/environment_variables":{"id":"bioinformatics/command_line/appendices/environment_variables","title":"Environment variables","description":"\\"Environment variables\\" are values that the system knows about and that you can access in your pipelines.","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/globbing":{"id":"bioinformatics/command_line/appendices/globbing","title":"Wildcards and \'globbing\'","description":"Globs are wildcard patterns that let the command-line match multiple files at once.","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/installing_wsl":{"id":"bioinformatics/command_line/appendices/installing_wsl","title":"Installing the Windows Subsystem for Linux","description":"The UNIX terminal for Windows requires Windows 10 or above. The full installation instructions are available on-line,","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/navigating_history":{"id":"bioinformatics/command_line/appendices/navigating_history","title":"Navigating history","description":"Remember that the shell keeps a command history.  You can see the history of","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/README":{"id":"bioinformatics/command_line/appendices/README","title":"Advanced topics","description":"This section contains a few pages on advanced topics that you might want to read once you have completed the main","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/table_of_commands":{"id":"bioinformatics/command_line/appendices/table_of_commands","title":"A table of useful commands","description":"Here are some useful commands - use man to get more information:","sidebar":"sidebar2"},"bioinformatics/command_line/appendices/where_are_the_commands":{"id":"bioinformatics/command_line/appendices/where_are_the_commands","title":"Where are the commands?","description":"A \'command\' is usually just a program on the filesystem somewhere.  A few commands, however are built-in to the command","sidebar":"sidebar2"},"bioinformatics/command_line/basics":{"id":"bioinformatics/command_line/basics","title":"Command line basics","description":"The command prompt","sidebar":"sidebar2"},"bioinformatics/command_line/filtering_files":{"id":"bioinformatics/command_line/filtering_files","title":"Filtering files: sort, uniq, cut, tr","description":"Pipelines are great, but you need some commands to put in them. Luckily bash contains lots of commands which can take","sidebar":"sidebar2"},"bioinformatics/command_line/help":{"id":"bioinformatics/command_line/help","title":"Getting help!","description":"Reading the man pages","sidebar":"sidebar2"},"bioinformatics/command_line/important_facts_about_filenames":{"id":"bioinformatics/command_line/important_facts_about_filenames","title":"Some important facts about file names","description":"It is worth remembering that:","sidebar":"sidebar2"},"bioinformatics/command_line/pipelines":{"id":"bioinformatics/command_line/pipelines","title":"I/O redirection and pipelining","description":"In programming there are three standard \'streams\' of communication","sidebar":"sidebar2"},"bioinformatics/command_line/questions":{"id":"bioinformatics/command_line/questions","title":"Challenge questions","description":"Now test yourself on the tutorial!  Here are a few tasks to try out.","sidebar":"sidebar2"},"bioinformatics/command_line/README":{"id":"bioinformatics/command_line/README","title":"Getting started with the command line","description":"Welcome!  The UNIX command-line interface (sometimes also called \u201ccommand prompt\u201d or \u201cterminal\u201d) is a way of","sidebar":"sidebar2"},"bioinformatics/command_line/standard_syntax":{"id":"bioinformatics/command_line/standard_syntax","title":"Standard command syntax","description":"Most commands follow the same basic syntax as ls. Here\'s a more advanced ls command:","sidebar":"sidebar2"},"bioinformatics/command_line/starting":{"id":"bioinformatics/command_line/starting","title":"Starting a terminal","description":"A terminal window is how you can access the command line.  To get started you need to open one!","sidebar":"sidebar2"},"bioinformatics/command_line/tour":{"id":"bioinformatics/command_line/tour","title":"Interacting with the filesystem","description":"Ok, so you have a terminal and you can run some commands... what now?","sidebar":"sidebar2"},"bioinformatics/command_line/working_with_files":{"id":"bioinformatics/command_line/working_with_files","title":"Working with files","description":"If you got this far, you\'ve had a first look at moving aronud the filesystem and manipulating files. But the files we","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/browsing":{"id":"bioinformatics/exploring_gene_annotations_in_bash/browsing","title":"Browsing the genome","description":"Can it be right that only a small fraction of these genes is coding? To confirm our results, let\u2019s look up on","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/counting":{"id":"bioinformatics/exploring_gene_annotations_in_bash/counting","title":"Counting genes","description":"Quit less if you are in it (by pressing q) and let\'s generate some basic statistics.","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/counting2":{"id":"bioinformatics/exploring_gene_annotations_in_bash/counting2","title":"Counting genes again","description":"So why are there 60,000 genes in the file - isn\'t that too many?","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/decompressing":{"id":"bioinformatics/exploring_gene_annotations_in_bash/decompressing","title":"Decompressing the file","description":"Does your file have .gz on the end of its filename?  And how big is it?  You can ls -lh to find out:","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/getting_started":{"id":"bioinformatics/exploring_gene_annotations_in_bash/getting_started","title":"Getting started","description":"To get started, if you [haven\'t","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/investigating":{"id":"bioinformatics/exploring_gene_annotations_in_bash/investigating","title":"Investigating specific genes","description":"Let\'s switch track and try to dig out info about a specific gene -","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/questions":{"id":"bioinformatics/exploring_gene_annotations_in_bash/questions","title":"Challenge questions","description":"The SNP rs601338 controls \u2019secretor status\u2019 (i.e. whether A/B","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/README":{"id":"bioinformatics/exploring_gene_annotations_in_bash/README","title":"Exploring gene annotations on the command-line","description":"Author: Gavin Band","sidebar":"sidebar2"},"bioinformatics/exploring_gene_annotations_in_bash/viewing":{"id":"bioinformatics/exploring_gene_annotations_in_bash/viewing","title":"Viewing gene records","description":"Use the less command to view the file:","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/data_types":{"id":"bioinformatics/introduction_to_R/data_types","title":"Some other data types","description":"Strings and numbers are just two of the data types R can work with.  What else?  Well, there are","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/functions":{"id":"bioinformatics/introduction_to_R/functions","title":"Functions","description":"So far so good - you can create variables and do computations.  To do really useful things, though, you need to use","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/fundamentals":{"id":"bioinformatics/introduction_to_R/fundamentals","title":"R as a calculator","description":"We\'ll assume you have started an R prompt - if not go and start one now.","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/getting_around":{"id":"bioinformatics/introduction_to_R/getting_around","title":"Getting around","description":"Before going further it\'s worth pointing out a few convenient features that R has to make your life easier.","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/help_going_wrong":{"id":"bioinformatics/introduction_to_R/help_going_wrong","title":"Help! it\'s not working!","description":"If something in the above examples isn\'t working - first of all, don\'t panic!","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/libaries":{"id":"bioinformatics/introduction_to_R/libaries","title":"Loading libraries","description":"So far everything we\'ve done has used \'base R\', but many useful functions can be found in libraries. In particular there","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/README":{"id":"bioinformatics/introduction_to_R/README","title":"Introduction to R","description":"Suggested prerequsite courses: Introduction to the Command Line, Programming Concepts.","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/summary":{"id":"bioinformatics/introduction_to_R/summary","title":"Summary","description":"In this tutorial we\'ve covered some basic ways of working with R, including","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/variables":{"id":"bioinformatics/introduction_to_R/variables","title":"Storing values in variables","description":"So far so good... we can compute with values, and \'vectors\' of values.  But to do real programming we will need to store","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/vectors":{"id":"bioinformatics/introduction_to_R/vectors","title":"Computing with lots of values at once","description":"One of the things you might have noticed is that when R prints out an answer, it puts [1]","sidebar":"sidebar2"},"bioinformatics/introduction_to_R/working_with_data":{"id":"bioinformatics/introduction_to_R/working_with_data","title":"Working with data","description":"Now we come to what R is really good at - loading, manipulating, and plotting data.","sidebar":"sidebar2"},"bioinformatics/making_an_R_package/fully_fledged":{"id":"bioinformatics/making_an_R_package/fully_fledged","title":"A fully-fledged R package","description":"Making a single file package like this is ok, but in R it\'s not a full package.  So if you are working in R, let\'s make one now.","sidebar":"sidebar2"},"bioinformatics/making_an_R_package/read_gff":{"id":"bioinformatics/making_an_R_package/read_gff","title":"The read_gff() function","description":"Before starting make sure you have a working read_gff() function.","sidebar":"sidebar2"},"bioinformatics/making_an_R_package/README":{"id":"bioinformatics/making_an_R_package/README","title":"Making an R or python package","description":"Welcome!  This tutorial assumes you have worked through the Analysing gene annotation in R or python tutorial first.  If so, you have a fully-fledged function called readgff() that can read data from a GFF file.","sidebar":"sidebar2"},"bioinformatics/making_an_R_package/single_file":{"id":"bioinformatics/making_an_R_package/single_file","title":"Taking the easy way","description":"The simplest way to make a \'package\' is to just put our function into a single file, like this:","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/004b_more_attributes":{"id":"bioinformatics/programming_with_gene_annotations3/004b_more_attributes","title":"Extracting more attributes","description":"If you followed the previous sections, you\'ll have a working read_gff() and you\'ll have run the test and seen the important message:","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/appendices/refactoring":{"id":"bioinformatics/programming_with_gene_annotations3/appendices/refactoring","title":"Refactor for better readability","description":"If you\'ve followed the main tutorial, your readgff() function will have","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/appendices/sqlite_access":{"id":"bioinformatics/programming_with_gene_annotations3/appendices/sqlite_access","title":"Reading results from a sqlite file in any language","description":"In the tutorial you wrote a command-line program to [convert the data into sqlite3","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/extracting_attributes":{"id":"bioinformatics/programming_with_gene_annotations3/extracting_attributes","title":"Extracting the ID and Parent attributes","description":"If you followed the previous page, you\'ll have a function called","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/Getting_started_writing_some_code":{"id":"bioinformatics/programming_with_gene_annotations3/Getting_started_writing_some_code","title":"A first go at parsing GFF","description":"This tutorial will lead you through a process of writing some code to load GFF data into either","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/Introduction":{"id":"bioinformatics/programming_with_gene_annotations3/Introduction","title":"Introduction","description":"Up to table of contents / Go to next page.","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/README":{"id":"bioinformatics/programming_with_gene_annotations3/README","title":"Analysing gene annotations in R or python","description":"Welcome! This tutorial has two aims. On one hand it is about programming, specifically the type of","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/recap":{"id":"bioinformatics/programming_with_gene_annotations3/recap","title":"Quick recap","description":"We followed a fairly straightforward path to write our read_gff() function.","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/testing_it_out":{"id":"bioinformatics/programming_with_gene_annotations3/testing_it_out","title":"Testing it out","description":"Hopefully you have now got a working function, read_gff() and got the extremely rewarding message:","sidebar":"sidebar2"},"bioinformatics/programming_with_gene_annotations3/What_gene_annotation_data_looks_like":{"id":"bioinformatics/programming_with_gene_annotations3/What_gene_annotation_data_looks_like","title":"What gene annotation data looks like","description":"Up to table of contents","sidebar":"sidebar2"},"bioinformatics/README":{"id":"bioinformatics/README","title":"Bioinformatics and programming","description":"Welcome!  To get started, choose your tutorial from the list below:","sidebar":"sidebar2"},"bioinformatics/tips_and_tricks/README":{"id":"bioinformatics/tips_and_tricks/README","title":"Tips and tricks","description":"Welcome! see the menu options on the left for some bioinformatics tips and tricks.","sidebar":"sidebar2"},"bioinformatics/tips_and_tricks/recursive_ftp":{"id":"bioinformatics/tips_and_tricks/recursive_ftp","title":"List contents of an FTP site","description":"The tool lftp can be used to quickly list contents of an FTP site. lftp calls itself a \'sophisticated file","sidebar":"sidebar2"},"Clinical variant interpretation/README":{"id":"Clinical variant interpretation/README","title":"Clinical variant interpretation","description":"Contributors: Nicky Whiffin"},"data_visualisation/data_visualisation_with_ggplot2/distribution":{"id":"data_visualisation/data_visualisation_with_ggplot2/distribution","title":"The distribution of GDP per capita","description":"In the rest of this tutorial we\'ll show various ways to create good-looking visualisations with","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/distribution2":{"id":"data_visualisation/data_visualisation_with_ggplot2/distribution2","title":"The distribution of GDP per capita, by continent","description":"Now we have another variable we want to add: continent. In a density plot, we can\'t add another","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/facetting":{"id":"data_visualisation/data_visualisation_with_ggplot2/facetting","title":"Facetting","description":"Right now our plot only shows one year of the data. What if we want to show all years at once? We\'d like a way to split the plot into several panes. This is what facets are designed for.","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/gapminder":{"id":"data_visualisation/data_visualisation_with_ggplot2/gapminder","title":"A first look at Gapminder data","description":"We\'ll use real population data from gapminder.org. To load it, load","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/gdp_life_expectancy":{"id":"data_visualisation/data_visualisation_with_ggplot2/gdp_life_expectancy","title":"Is gdp related to life expectancy?","description":"In the above we\'ve been looking at the distribution of gdp across all years in the data.","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/labelling":{"id":"data_visualisation/data_visualisation_with_ggplot2/labelling","title":"Labelling points","description":"If you look back at [this plot of GDP versus life","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/other_plots":{"id":"data_visualisation/data_visualisation_with_ggplot2/other_plots","title":"Other plots - and a challenge","description":"There are of course lots of other ways we could visualise this data -","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/README":{"id":"data_visualisation/data_visualisation_with_ggplot2/README","title":"More data visualisation in R","description":"Author: Samvida Venkatesh.","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/reordering":{"id":"data_visualisation/data_visualisation_with_ggplot2/reordering","title":"Reordering values","description":"To make this plot easier to read, let\'s order the plots by median GDP per capita.","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/saving":{"id":"data_visualisation/data_visualisation_with_ggplot2/saving","title":"Saving your figures","description":"To make sure you don\'t lose all your hard work - let\'s save the plot to a pdf. (You can also save to tiff, png, etc.,","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/summary_statistics":{"id":"data_visualisation/data_visualisation_with_ggplot2/summary_statistics","title":"Adding summary statistics","description":"These violins are essentially vertically-plotted, back-to-back histograms. They are pretty useful","sidebar":"sidebar3"},"data_visualisation/data_visualisation_with_ggplot2/violins":{"id":"data_visualisation/data_visualisation_with_ggplot2/violins","title":"Bring on the violins","description":"Density plots like this look kindof nice, but can be difficult to interpret with many groups","sidebar":"sidebar3"},"data_visualisation/README":{"id":"data_visualisation/README","title":"Data visualisation tutorials","description":"Welcome!  To get started, choose your tutorial from the list below:","sidebar":"sidebar3"},"data_visualisation/visualising_rna-seq_data/appendices/getting_session_info":{"id":"data_visualisation/visualising_rna-seq_data/appendices/getting_session_info","title":"Getting information about your R session","description":"The sessionInfo() command is useful if you want to see all detailed information about your current R installation:","sidebar":"sidebar3"},"data_visualisation/visualising_rna-seq_data/getting_setup":{"id":"data_visualisation/visualising_rna-seq_data/getting_setup","title":"Getting setup","description":"Loading libraries","sidebar":"sidebar3"},"data_visualisation/visualising_rna-seq_data/README":{"id":"data_visualisation/visualising_rna-seq_data/README","title":"Visualise RNA-seq data in R","description":"Author: Kiki Cano-Gamez.","sidebar":"sidebar3"},"data_visualisation/visualising_rna-seq_data/visualising":{"id":"data_visualisation/visualising_rna-seq_data/visualising","title":"Visualising the data","description":"Visualisation is usually the first step when exploring a data set, since it enables us to assess data quality and identify any major patterns. Which techniques and types of graphs we use for visualisation is determined by the nature of our data. For example, how many variables do we have? What are we interested in assessing?","sidebar":"sidebar3"},"genome_wide_association_studies/file_formats/gcta":{"id":"genome_wide_association_studies/file_formats/gcta","title":"GCTA file formats","description":".mlma file","sidebar":"sidebar8"},"genome_wide_association_studies/file_formats/oxford":{"id":"genome_wide_association_studies/file_formats/oxford","title":"\'Oxford\' file formats format","description":".gen file","sidebar":"sidebar8"},"genome_wide_association_studies/file_formats/plink":{"id":"genome_wide_association_studies/file_formats/plink","title":"PLINK file formats","description":"Data in binary PLINK format, such as that provided for this tutorial, consists of a set of three files each containing information about the genotyping data and the samples included in the dataset. A brief description of each is provided below. More information on file formats can be found on the PLINK website","sidebar":"sidebar8"},"genome_wide_association_studies/file_formats/README":{"id":"genome_wide_association_studies/file_formats/README","title":"GWAS file formats","description":"See the sidebar links for file format descriptions for various GWAS file formats.","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies":{"id":"genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies","title":"Plink warm-up","description":"Plink is a tool for analysis of genome-wide genotype data that supports many functions including those related to data","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/getting_setup":{"id":"genome_wide_association_studies/genome_wide_association_analysis/getting_setup","title":"Getting setup","description":"In this practical we will run command-line programs to perform the analyses, and R to interpret them. If you haven\'t","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink":{"id":"genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink","title":"A toy GWAS","description":"Before we get started properly let\'s run a little toy example.","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/gwas_association":{"id":"genome_wide_association_studies/genome_wide_association_analysis/gwas_association","title":"A real (simulated) GWAS: testing for association","description":"Now we will test for association between the SNPs in our dataset and the disease outcome.","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control":{"id":"genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control","title":"A real (simulated) GWAS: quality control","description":"We will now work with a set of data files containing many SNPs from chromosome 19, which have been genotyped on a set of individuals who were infected with norovirus (disease cases) and a set of population controls","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/hitplots":{"id":"genome_wide_association_studies/genome_wide_association_analysis/hitplots","title":"Making a regional association plot","description":"In this part of the practical, we will hone our R plotting skills to make beautiful regional association plots (\u2018hitplots\u2019) of","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/population_structure":{"id":"genome_wide_association_studies/genome_wide_association_analysis/population_structure","title":"Controlling for population structure","description":"We have seen how applying appropriate quality control filters to our data eliminated many false positives, but a systematic","sidebar":"sidebar8"},"genome_wide_association_studies/genome_wide_association_analysis/README":{"id":"genome_wide_association_studies/genome_wide_association_analysis/README","title":"Genome-wide association study example: a case-control trait","description":"Welcome!  This practical will introduce you to analyzing Genome Wide Association Study (GWAS) datasets using a program","sidebar":"sidebar8"},"genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping":{"id":"genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping","title":"Using FINEMAP to fine-map associations","description":"Up to the table of contents / Back to the meta-analysis section","sidebar":"sidebar8"},"genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction":{"id":"genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction","title":"Meta-analysis and fine-mapping practical","description":"Up to the table of contents / Forward to the meta-analysis section","sidebar":"sidebar8"},"genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis":{"id":"genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis","title":"Meta-analysing two studies.","description":"Up to the table of contents / Back to the Introduction / Forward to the fine-mapping section","sidebar":"sidebar8"},"genome_wide_association_studies/meta-analysis_and_fine-mapping/README":{"id":"genome_wide_association_studies/meta-analysis_and_fine-mapping/README","title":"Meta-analysis and fine-mapping tutorial","description":"Contributors: Luke Jostins, Gavin Band","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/effect_sizes":{"id":"genome_wide_association_studies/quantitative_trait_gwas/effect_sizes","title":"Assessing effect sizes","description":"Looking at the genomic region around our main peak, one of the genes in this region is fucosyltransferase 2 (FUT2).","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/getting_the_data":{"id":"genome_wide_association_studies/quantitative_trait_gwas/getting_the_data","title":"Getting the data","description":"You will be using four sets of files for this exercise. It is recommended that you download these files to your working directory prior to starting the exercise Genotype data files.  To get started, make a new directory to run the tutorial in and cd into it:","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/getting_the_software":{"id":"genome_wide_association_studies/quantitative_trait_gwas/getting_the_software","title":"Getting the software","description":"In this tutorial we will use two pieces of software designed for GWAS analysis:","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/imputed_data":{"id":"genome_wide_association_studies/quantitative_trait_gwas/imputed_data","title":"Part 5: Imputation","description":"Genotype imputation allows us to test associations at ungenotyped markers, increase genotyping density for fine-mapping","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/independent_signals":{"id":"genome_wide_association_studies/quantitative_trait_gwas/independent_signals","title":"Appendix: Looking for independent signals","description":"It is of course possible that there are multiple independent signals within a single region. To test if this is the","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/interpretation":{"id":"genome_wide_association_studies/quantitative_trait_gwas/interpretation","title":"Part 6: Interpreting GWAS signals","description":"Locating significant signals and defining regions of interest","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/phenotype_qc":{"id":"genome_wide_association_studies/quantitative_trait_gwas/phenotype_qc","title":"Part 2: Transforming the phenotype data","description":"Most commonly-used GWAS software implement","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/population_structure":{"id":"genome_wide_association_studies/quantitative_trait_gwas/population_structure","title":"Part 4: Working with population substructure and multi-ethnic populations","description":"Our analysis up to now has been carried out under the assumption that response to Norovirus is standard across everyone,","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes":{"id":"genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes","title":"Part 1: Quality control of genotype data","description":"Calculating summary statistics for the data","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/README":{"id":"genome_wide_association_studies/quantitative_trait_gwas/README","title":"Genome-wide association with quantitative antibody levels","description":"Author: Amanda Chong","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/relatedness":{"id":"genome_wide_association_studies/quantitative_trait_gwas/relatedness","title":"Estimating relatedness","description":"Now let\'s estimate the relatedness between different individuals in the dataset.","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/sample_relatedness":{"id":"genome_wide_association_studies/quantitative_trait_gwas/sample_relatedness","title":"Aside: excluding related samples","description":"Note. This section is optional. We won\'t actually use it in the assocation test for this tutorial.","sidebar":"sidebar8"},"genome_wide_association_studies/quantitative_trait_gwas/testing_for_association":{"id":"genome_wide_association_studies/quantitative_trait_gwas/testing_for_association","title":"Part 3: Running a simple regression analysis","description":"Now that you have a good idea of what your data looks like, we are going to run a basic linear regression using filtered","sidebar":"sidebar8"},"genome_wide_association_studies/README":{"id":"genome_wide_association_studies/README","title":"Genome-wide association analysis","description":"Welcome!  In this folder are a set of practical tutorial which will get you started analysing genome-wide association study data.","sidebar":"sidebar8"},"genome_wide_association_studies/slides/README":{"id":"genome_wide_association_studies/slides/README","title":"README","description":"Slides from these sessions can be found here:"},"LICENSE":{"id":"LICENSE","title":"LICENSE","description":"MIT License"},"next_generation_sequencing/building_an_ngs_pipeline/README":{"id":"next_generation_sequencing/building_an_ngs_pipeline/README","title":"The NGS analysis pipeline challenge!","description":"Welcome!  Go to the introduction to get started."},"overview":{"id":"overview","title":"Overview of the tutorials","description":"Welcome!  This folder contains course materials (practicals, notes and datasets) that were developed for the taught"},"population_genetics/admixture_and_biobanks/admixture":{"id":"population_genetics/admixture_and_biobanks/admixture","title":"ADMIXTURE analysis","description":"ADMIXTURE is among many other software (e.g. STRUCTURE) that perform clustering based on genotyping data to infer","sidebar":"sidebar7"},"population_genetics/admixture_and_biobanks/fst":{"id":"population_genetics/admixture_and_biobanks/fst","title":"Fst","description":"$F_$ is a statistical measure used to quantify the degree of genetic differentiation or genetic structure among populations.","sidebar":"sidebar7"},"population_genetics/admixture_and_biobanks/getting_setup":{"id":"population_genetics/admixture_and_biobanks/getting_setup","title":"Getting setup","description":"Getting the data","sidebar":"sidebar7"},"population_genetics/admixture_and_biobanks/pca":{"id":"population_genetics/admixture_and_biobanks/pca","title":"Principal component analysis of the 1000 Genomes Project","description":"The example is based on a subset of genotype data of common variant included in the 1000 Genome Project (chromosome 21","sidebar":"sidebar7"},"population_genetics/admixture_and_biobanks/README":{"id":"population_genetics/admixture_and_biobanks/README","title":"Population structure and admixture","description":"Authors: Yang Luo and Jason Torres","sidebar":"sidebar7"},"population_genetics/plotting_haplotypes/README":{"id":"population_genetics/plotting_haplotypes/README","title":"Visualising haplotypes in R","description":"In this page you will load up some real haplotype data (from the 1000 Genomes Project) and plot it in R.","sidebar":"sidebar7"},"population_genetics/plotting_haplotypes/tbc":{"id":"population_genetics/plotting_haplotypes/tbc","title":"tbc","description":"","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/computing_PCs":{"id":"population_genetics/principal_components_analysis/computing_PCs","title":"Computing principal components","description":"If you\'ve reached this page, you should now have downloaded the data,","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/getting_setup":{"id":"population_genetics/principal_components_analysis/getting_setup","title":"Getting set up","description":"Genome-wide genetic datasets are typically too big to load in R, so to compute PCs across them you typically need some dedicated software. In this tutorial we\'ll use \'plink\' to compute PCs.","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/global_analysis":{"id":"population_genetics/principal_components_analysis/global_analysis","title":"Plotting samples against a global reference panel","description":"Up to the table of contents / Back to the PC computation step","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/ld_pruning":{"id":"population_genetics/principal_components_analysis/ld_pruning","title":"Removing highly-correlated SNPs","description":"If you followed the population genetics simulation tutorial, you\'ll know that","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/overview":{"id":"population_genetics/principal_components_analysis/overview","title":"Overview of the practical","description":"Up to the table of contents / Back to the setup page / Forward to the page on LD pruning","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/README":{"id":"population_genetics/principal_components_analysis/README","title":"Principal components analysis","description":"Welcome!  In this practical we will do principal component analysis (PCA), one of the fundamental tools of population","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/relatedness_pruning":{"id":"population_genetics/principal_components_analysis/relatedness_pruning","title":"Removing closely-related samples","description":"Up to the table of contents - Back to the page on LD pruning - Forward to compute the PCs","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/simulated_pca_example":{"id":"population_genetics/principal_components_analysis/simulated_pca_example","title":"A warm-up example","description":"Before running the main practical, let\'s first see how PCa works by using it to recover some simulated structure from a matrix in R.","sidebar":"sidebar7"},"population_genetics/principal_components_analysis/the_maths":{"id":"population_genetics/principal_components_analysis/the_maths","title":"An aside on the maths","description":"The maths of PCA (in the way usually applied) works like this. Suppose $X$ is a big matrix of genotypes, with $L$ rows","sidebar":"sidebar7"},"population_genetics/README":{"id":"population_genetics/README","title":"Population genetics","description":"In this folder are:","sidebar":"sidebar7"},"population_genetics/simulation/mutations":{"id":"population_genetics/simulation/mutations","title":"Adding mutations","description":"So, genetic drift causes loss of haplotypes from the population and generates LD.","sidebar":"sidebar7"},"population_genetics/simulation/popgen_simulation":{"id":"population_genetics/simulation/popgen_simulation","title":"Genetic drift","description":"The Wright-Fisher model is about the simplest population-genetic model there is.  It consists of:","sidebar":"sidebar7"},"population_genetics/simulation/README":{"id":"population_genetics/simulation/README","title":"Simulating populations","description":"In this practical we will implement a simple Wright-Fisher population genetics model, and couple","sidebar":"sidebar7"},"population_genetics/simulation/statistics":{"id":"population_genetics/simulation/statistics","title":"Some population-genetic statistics","description":"If you followed the basic simulation section you\'ll know that random inheritance","sidebar":"sidebar7"},"prerequisites/CONDA":{"id":"prerequisites/CONDA","title":"Setting up Conda","description":"One of the easiest ways to set up your environment that works across platforms is to use conda.","sidebar":"sidebar1"},"prerequisites/editor":{"id":"prerequisites/editor","title":"Getting a text editor","description":"Whatever you do, you\'re going to want a good text editor.  There are many choices -","sidebar":"sidebar1"},"prerequisites/julia":{"id":"prerequisites/julia","title":"Julia","description":"Julia is an interesting language that is pretty good for many data analysis and computational","sidebar":"sidebar1"},"prerequisites/Jupyterlab":{"id":"prerequisites/Jupyterlab","title":"Using R, python and julia with JupyterLab","description":"The plain command-line interpreters for R, python and julia are useful, but for data analysis work you may want","sidebar":"sidebar1"},"prerequisites/Other_languages/javascript":{"id":"prerequisites/Other_languages/javascript","title":"Javascript","description":"You don\'t need to install javascript - you have it already in your browser. To have a go at using","sidebar":"sidebar1"},"prerequisites/Other_languages/README":{"id":"prerequisites/Other_languages/README","title":"Other languages","description":"There are lots of other programming languages out there - useful for different tasks. Here are some:","sidebar":"sidebar1"},"prerequisites/python":{"id":"prerequisites/python","title":"Python","description":"You shouldn\'t need to install python - both your base system, and your conda environment should have a version of python","sidebar":"sidebar1"},"prerequisites/R":{"id":"prerequisites/R","title":"R and RStudio","description":"A great deal of data analysis work is done in R - you\'ll need this","sidebar":"sidebar1"},"prerequisites/README":{"id":"prerequisites/README","title":"Getting your environment setup","description":"The tutorials found here involve running bioinformatics tools and writing computer code. You will","sidebar":"sidebar1"},"prerequisites/tidyverse":{"id":"prerequisites/tidyverse","title":"The R `tidyverse`","description":"Base R is ok but it can be a bit... clunky. For many of the tutorials we wil use a set of","sidebar":"sidebar1"},"prerequisites/UNIX":{"id":"prerequisites/UNIX","title":"Setting up a UNIX terminal","description":"You will be doing a lot of work in the UNIX terminal - this is a command-line interface.","sidebar":"sidebar1"},"programming/pipelining_with_snakemake/prerequisites":{"id":"programming/pipelining_with_snakemake/prerequisites","title":"Getting set up","description":"To get started there are three steps.","sidebar":"sidebar5"},"programming/pipelining_with_snakemake/README":{"id":"programming/pipelining_with_snakemake/README","title":"Introduction to snakemake","description":"Welcome!  In this page you will see a brief introduction to snakemake.","sidebar":"sidebar5"},"programming/pipelining_with_snakemake/snakefile3":{"id":"programming/pipelining_with_snakemake/snakefile3","title":"snakefile3","description":"---","sidebar":"sidebar5"},"programming/pipelining_with_snakemake/using_snakemake":{"id":"programming/pipelining_with_snakemake/using_snakemake","title":"Using snakemake","description":"(This page is under construction).","sidebar":"sidebar5"},"recent_contributors":{"id":"recent_contributors","title":"Recent contributors","description":"We are grateful to the following people who have contributed to WHG training in recent years:"},"rna_seq_and_other_omics/intro_to_rna_seq_analysis/Data_preprocessing":{"id":"rna_seq_and_other_omics/intro_to_rna_seq_analysis/Data_preprocessing","title":"Data preprocessing","description":"We start from the raw count table, which holds the number of reads aligning to each gene in each sample.","sidebar":"sidebar10"},"rna_seq_and_other_omics/intro_to_rna_seq_analysis/Diff_exp_genes":{"id":"rna_seq_and_other_omics/intro_to_rna_seq_analysis/Diff_exp_genes","title":"Identifying differentially expressed genes","description":"First, we need to create a \\"design matrix\\" capturing the experimental design - this has one row per sample and experimental groups coded as 0 or 1 in the columns. We can add covariates if needed but for now we will do a simple comparison of the nutrient-rich vs starved conditions (n=6 in each group). We make use of the conds factor that we created earlier.","sidebar":"sidebar10"},"rna_seq_and_other_omics/intro_to_rna_seq_analysis/README":{"id":"rna_seq_and_other_omics/intro_to_rna_seq_analysis/README","title":"RNA-Seq data analysis","description":"In this tutorial we will demonstrate how to analyse RNA-Seq data using a published dataset (https://pmc.ncbi.nlm.nih.gov/articles/PMC10050925/), which investigated the transcriptional effects of cell starvation in HeLa cells. Twelve samples were profiled and we will generate a list of differentially expressed (DE) genes between the high-nutrient and starvation conditions.","sidebar":"sidebar10"},"rna_seq_and_other_omics/README":{"id":"rna_seq_and_other_omics/README","title":"RNA-seq and other omics","description":"Welcome!  Use the sidebar to find practicals.","sidebar":"sidebar10"},"sequence_data_analysis/building_an_ngs_pipeline/Introduction":{"id":"sequence_data_analysis/building_an_ngs_pipeline/Introduction","title":"Introduction","description":"In this tutorial you will bring a set of FASTQ files","sidebar":"sidebar9"},"sequence_data_analysis/building_an_ngs_pipeline/pipeline":{"id":"sequence_data_analysis/building_an_ngs_pipeline/pipeline","title":"The pipeline","description":"Overview","sidebar":"sidebar9"},"sequence_data_analysis/building_an_ngs_pipeline/README":{"id":"sequence_data_analysis/building_an_ngs_pipeline/README","title":"A sequence data pipelining challenge!","description":"Welcome!  Go to the introduction to get started.","sidebar":"sidebar9"},"sequence_data_analysis/building_an_ngs_pipeline/tips_and_tricks":{"id":"sequence_data_analysis/building_an_ngs_pipeline/tips_and_tricks","title":"Tips and tricks","description":"Here is some guidance to help you write your pipeline. Click the links to jump to the relevant","sidebar":"sidebar9"},"sequence_data_analysis/Getting started with snakemake/README":{"id":"sequence_data_analysis/Getting started with snakemake/README","title":"Getting started with snakemake","description":"This is a quick snakemake example that uses bwa mem to align a few reads."},"sequence_data_analysis/IGV/Exploring_reads_in_IGV":{"id":"sequence_data_analysis/IGV/Exploring_reads_in_IGV","title":"Viewing reads - the basics","description":"You should now have load some data into IGV (if not, go and do that now.)","sidebar":"sidebar9"},"sequence_data_analysis/IGV/hard_to_map":{"id":"sequence_data_analysis/IGV/hard_to_map","title":"Hard-to-align regions","description":"Question. If you scroll around a bit, after a while you\'ll come across regions where no reads (or very few reads)","sidebar":"sidebar9"},"sequence_data_analysis/IGV/indels":{"id":"sequence_data_analysis/IGV/indels","title":"Looking at insertions and deletions","description":"The variants above are single nucleotide polymorphisms (SNPs) - or rather multiple nucleotide","sidebar":"sidebar9"},"sequence_data_analysis/IGV/README":{"id":"sequence_data_analysis/IGV/README","title":"Exploring Reads in IGV","description":"Contributors:  Gavin Band, Annie Forster","sidebar":"sidebar9"},"sequence_data_analysis/IGV/SNPs":{"id":"sequence_data_analysis/IGV/SNPs","title":"Looking at SNPs","description":"Let\'s try looking some SNPs.","sidebar":"sidebar9"},"sequence_data_analysis/IGV/SVs":{"id":"sequence_data_analysis/IGV/SVs","title":"Examining larger structural variants","description":"Let\'s have a look at a larger structural variant.  Zoom your IGV to the region around the gene Pf3D71127000.  And zoom out a bit so you can see the whole region including the two flanking genes.","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads","title":"Step 3: Aligning reads","description":"Most analyses of short-read sequence data are based on the basic paradigm made up of two steps:","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions","title":"Challenge questions","description":"We have three challenges.:","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation","title":"Bonus tutorial: how accurate is Illumina sequencing?","description":"Note. Only do this section if you have plenty of time left, and/or have already read the","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates","title":"Appendix 2: understanding sequence duplication levels","description":"Duplicates arise naturally in sequencing from random fragments that just happen to have the same breakpoints. However,","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/getting_the_data":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/getting_the_data","title":"Getting the data","description":"You\'ll also need some data - to get this,  download the tarball from [this","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs","title":"Step 1: What\'s in the fastq files?","description":"Using UNIX to inspect the fastqs","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline","title":"Practical outline","description":"Overview","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites","title":"Getting the software","description":"To run this tutorial you will need some software.  This page will get you set up.","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control","title":"Step 2: Inspecting sequence data quality","description":"There are a set of analyses that can be carried out directly on a FASTQ file to assess the quality","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming","title":"Appendix 3: read trimming","description":"Read trimming refers to the process of removing trailing parts of the reads before analysis for various reasons including","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/README":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/README","title":"Analysing paired-end sequencing data","description":"In this tutorial we will demonstrate a basic pipeline for analysing paired-end short-read genomic","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq","title":"Appendix 4: RNA-seq links","description":"We\'re not covering RNA-seq on this page, but see this comprehensive overview for an introduction to all","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory","title":"Appendix 1: Paired-end sequencing in theory and practice","description":"What paired-end sequence data should look like","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links","title":"Useful links","description":"* Illumina\'s page on base quality scores.","sidebar":"sidebar9"},"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments":{"id":"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments","title":"Step 4: Viewing alignments","description":"If you got this far, you should have a file called QG0033-C.bam containing aligned,","sidebar":"sidebar9"},"sequence_data_analysis/README":{"id":"sequence_data_analysis/README","title":"Next-generation sequencing","description":"* Analyse some Sanger sequencing data tutorial.","sidebar":"sidebar9"},"sequence_data_analysis/sanger_sequence_data/introduction":{"id":"sequence_data_analysis/sanger_sequence_data/introduction","title":"Viewing Sanger trace files","description":"If you got here, you should have downloaded a folder of Sanger sequence data.  (If not please [go and do that","sidebar":"sidebar9"},"sequence_data_analysis/sanger_sequence_data/README":{"id":"sequence_data_analysis/sanger_sequence_data/README","title":"Analysing Sanger sequence data","description":"Welcome!  In this tutorial we will conduct a basic analysis of some Sanger sequence data. The data comes from a PCR of","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_a":{"id":"sequence_data_analysis/sightseeing_tour/dataset_a","title":"Dataset A: Illumina paired-end","description":"File URL:","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_b":{"id":"sequence_data_analysis/sightseeing_tour/dataset_b","title":"Datasets B and C: long reads","description":"Dataset B URL:","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_c":{"id":"sequence_data_analysis/sightseeing_tour/dataset_c","title":"DNA modifications","description":"One of the amazing things about long-read sequencing is that it can measure DNA modifications, as well as just DNA bases.","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_d":{"id":"sequence_data_analysis/sightseeing_tour/dataset_d","title":"Dataset D: Illumina RNA-seq","description":"Dataset URLs:","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_e":{"id":"sequence_data_analysis/sightseeing_tour/dataset_e","title":"Dataset E: ATAC-seq","description":"Dataset E URL:","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_f":{"id":"sequence_data_analysis/sightseeing_tour/dataset_f","title":"Dataset E: 10X linked-read sequencing","description":"Dataset link:","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/dataset_g":{"id":"sequence_data_analysis/sightseeing_tour/dataset_g","title":"New! Dataset G: long-read RNA-seq","description":"Dataset G URL:","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/quiz":{"id":"sequence_data_analysis/sightseeing_tour/quiz","title":"Play \'guess the dataset\'","description":"Ok now you are expert in different types of data - it\'s time to test yourself.","sidebar":"sidebar9"},"sequence_data_analysis/sightseeing_tour/README":{"id":"sequence_data_analysis/sightseeing_tour/README","title":"IGV sightseeing tour","description":"In this section we will have a quick tour through several types of genomic sequencing, using data from a single sample.","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/Challenge_questions":{"id":"sequence_data_analysis/variant_calling_and_imputation/Challenge_questions","title":"Challenge questions on imputation","description":"Imputation $r^2$","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/Imputation":{"id":"sequence_data_analysis/variant_calling_and_imputation/Imputation","title":"Imputing a set of microarray data","description":"Ok, so we\'ve called genetic variants, we\'ve [QCd","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/Phasing":{"id":"sequence_data_analysis/variant_calling_and_imputation/Phasing","title":"Phasing the variant calls","description":"If you got this far you should have a robust, well-QCd set of variants. But we\'re not done! Humans","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/Prerequisites":{"id":"sequence_data_analysis/variant_calling_and_imputation/Prerequisites","title":"Getting the data","description":"To get the data download and unzip this tarball:","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/README":{"id":"sequence_data_analysis/variant_calling_and_imputation/README","title":"Variant calling, phasing and imputation","description":"This morning we focussed on quality control, aligning, and inspecting some sequence data for a single sample.","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/Variant_calling":{"id":"sequence_data_analysis/variant_calling_and_imputation/Variant_calling","title":"Variant calling with bcftools","description":"First let\'s see how to use a simple pipeline to identify genetic variants","sidebar":"sidebar9"},"sequence_data_analysis/variant_calling_and_imputation/Variant_quality_control":{"id":"sequence_data_analysis/variant_calling_and_imputation/Variant_quality_control","title":"Variant quality control","description":"For variant quality control we\'ll start with the output of bcftools call and use the various INFO","sidebar":"sidebar9"},"statistical_modelling/bayesian_statistics/figures/README":{"id":"statistical_modelling/bayesian_statistics/figures/README","title":"README","description":"Directory for images."},"statistical_modelling/bayesian_statistics/README":{"id":"statistical_modelling/bayesian_statistics/README","title":"Bayesian statistics","description":"Welcome!  This folder contains tutorials on Bayesian modelling and Markov Chain Monte Carlo (MCMC)."},"statistical_modelling/distributions_cheatsheet":{"id":"statistical_modelling/distributions_cheatsheet","title":"Distributions cheatsheet","description":"See below for a cheat-sheet on some commonly used probability distributions, or download directly.","sidebar":"sidebar6"},"statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm":{"id":"statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm","title":"Modelling CNVs using a HMM","description":"To run this part of the practical, you first need to have followed the warm-up tutorial and computed a set of log-likelihoods for each sample at each site in the relevant dataset.","sidebar":"sidebar6"},"statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup":{"id":"statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup","title":"Warmup: examining read coverage in a region","description":"Welcome! The practical studies a region of the human genome on chromosome 4 that is known to","sidebar":"sidebar6"},"statistical_modelling/Hidden_markov_models/modelling_copy_number_variation":{"id":"statistical_modelling/Hidden_markov_models/modelling_copy_number_variation","title":"Maximum likelihood copy number inference","description":"Remember that the \'likelihood function\' should compute the \\"probability of the data given the parameters of interest\\".","sidebar":"sidebar6"},"statistical_modelling/Hidden_markov_models/modelling_copy_number_variation_bayes":{"id":"statistical_modelling/Hidden_markov_models/modelling_copy_number_variation_bayes","title":"Bayesian copy number inference","description":"Taking the maximum likelihood approach as we [did a moment","sidebar":"sidebar6"},"statistical_modelling/Hidden_markov_models/README":{"id":"statistical_modelling/Hidden_markov_models/README","title":"Glycophorin CNV calling tutorial","description":"Welcome! In this tutorial you will build a hidden markov model to model sequence read coverage","sidebar":"sidebar6"},"statistical_modelling/introduction/allele_frequencies":{"id":"statistical_modelling/introduction/allele_frequencies","title":"Estimating allele frequencies","description":"Let\'s use Bayes to estimate allele frequencies - quantifying our uncertainty - for a couple of important","sidebar":"sidebar6"},"statistical_modelling/introduction/allele_frequencies_2":{"id":"statistical_modelling/introduction/allele_frequencies_2","title":"A bayesian forest plot","description":"The plot we just made is ok, but surely visually pretty wasteful.","sidebar":"sidebar6"},"statistical_modelling/introduction/appendix/README":{"id":"statistical_modelling/introduction/appendix/README","title":"Appendix: solutions","description":"Here is a solution to the forest plot question."},"statistical_modelling/introduction/bayes":{"id":"statistical_modelling/introduction/bayes","title":"The most important formula in science","description":"The most important formula is... Bayes theorem!","sidebar":"sidebar6"},"statistical_modelling/introduction/bayesian_update":{"id":"statistical_modelling/introduction/bayesian_update","title":"The bayesian update","description":"Now let\'s consider a more complex situation.","sidebar":"sidebar6"},"statistical_modelling/introduction/challenge":{"id":"statistical_modelling/introduction/challenge","title":"The normal approximation","description":"As the last part of this tutorial, I wanted to mention an approximation that is very very important in","sidebar":"sidebar6"},"statistical_modelling/introduction/covid":{"id":"statistical_modelling/introduction/covid","title":"What to make of a COVID test","description":"Let\'s apply Bayes formula to this question:","sidebar":"sidebar6"},"statistical_modelling/introduction/diagram":{"id":"statistical_modelling/introduction/diagram","title":"Where statistics fits in","description":"In my mind, science looks something like this:","sidebar":"sidebar6"},"statistical_modelling/introduction/distributions":{"id":"statistical_modelling/introduction/distributions","title":"Distributions","description":"In statistics, the way uncertainty is accounted for is through  probability distributions.","sidebar":"sidebar6"},"statistical_modelling/introduction/drug_example":{"id":"statistical_modelling/introduction/drug_example","title":"Toy example 1: a drug trial","description":"Acme Inc. has run a trial of some new treatments against a chronic, currently untreatable condition. They ascertained affected people and either gave them a placebo, or one of two treatments.  They then followed up with each participant after a year.","sidebar":"sidebar6"},"statistical_modelling/introduction/quantiles_and_pvalues":{"id":"statistical_modelling/introduction/quantiles_and_pvalues","title":"Quantiles, p-values and all that","description":"Consider a variable $X$ that is normally distributed - here is a picture of its distribution:","sidebar":"sidebar6"},"statistical_modelling/introduction/README":{"id":"statistical_modelling/introduction/README","title":"From basics to Bayes","description":"These pages still under construction.","sidebar":"sidebar6"},"statistical_modelling/introduction/reasoning":{"id":"statistical_modelling/introduction/reasoning","title":"Statistics is reasoning with uncertainty","description":"I read the above title on the internet.  I","sidebar":"sidebar6"},"statistical_modelling/introduction/some_distributions":{"id":"statistical_modelling/introduction/some_distributions","title":"Some useful probability distributions","description":"In practice in statistics we often use a set of well-known distributions.","sidebar":"sidebar6"},"statistical_modelling/introduction/summarising_distributions":{"id":"statistical_modelling/introduction/summarising_distributions","title":"Distribution summaries","description":"We can also handle distributions by creating some single-number summaries of the distributions.","sidebar":"sidebar6"},"statistical_modelling/notes/computing_pvalues":{"id":"statistical_modelling/notes/computing_pvalues","title":"Asymptotics and  P-values","description":"Much of classical statistics relies on \'asymptotics\', which means statistical behaviour when data","sidebar":"sidebar6"},"statistical_modelling/probability_cheatsheet":{"id":"statistical_modelling/probability_cheatsheet","title":"Probability cheatsheet","description":"Welcome! See below for the probability cheatsheet, or download directly.","sidebar":"sidebar6"},"statistical_modelling/README":{"id":"statistical_modelling/README","title":"Statistical modelling","description":"Welcome!  This folder contains tutorials on statistical modelling.","sidebar":"sidebar6"},"statistical_modelling/regression_modelling/causal_graphs":{"id":"statistical_modelling/regression_modelling/causal_graphs","title":"Causal thinking","description":"In the linear regression example we saw some pretty odd behaviour. A model fit of genotype on expression","sidebar":"sidebar6"},"statistical_modelling/regression_modelling/interpreting_p_values":{"id":"statistical_modelling/regression_modelling/interpreting_p_values","title":"How to interpret the P-value","description":"So you\'ve done a statistical test, got an estimate, a standard error and a P-value. How do you know when there\'s enough evidence","sidebar":"sidebar6"},"statistical_modelling/regression_modelling/linear_regression_1":{"id":"statistical_modelling/regression_modelling/linear_regression_1","title":"A linear regression example","description":"Prerequisites","sidebar":"sidebar6"},"statistical_modelling/regression_modelling/linear_regression_cheatsheet":{"id":"statistical_modelling/regression_modelling/linear_regression_cheatsheet","title":"Linear regression cheatsheet","description":"See below for the probability cheatsheet, or download directly.","sidebar":"sidebar6"},"statistical_modelling/regression_modelling/logistic_regression_1":{"id":"statistical_modelling/regression_modelling/logistic_regression_1","title":"A logistic regression example","description":"Before moving on let\'s try a different type of regression: logistic regression. Unlike linear regression, which models a","sidebar":"sidebar6"},"statistical_modelling/regression_modelling/README":{"id":"statistical_modelling/regression_modelling/README","title":"Modelling statistical association with regression","description":"Welcome!  This folder contains tutorials on modelling using regression.","sidebar":"sidebar6"}}}')}}]);