"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[567],{3905:(e,t,n)=>{n.d(t,{Zo:()=>p,kt:()=>g});var a=n(67294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},r=Object.keys(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var l=a.createContext({}),c=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},p=function(e){var t=c(e.components);return a.createElement(l.Provider,{value:t},e.children)},d="mdxType",m={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},u=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,r=e.originalType,l=e.parentName,p=s(e,["components","mdxType","originalType","parentName"]),d=c(n),u=i,g=d["".concat(l,".").concat(u)]||d[u]||m[u]||r;return n?a.createElement(g,o(o({ref:t},p),{},{components:n})):a.createElement(g,o({ref:t},p))}));function g(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=n.length,o=new Array(r);o[0]=u;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[d]="string"==typeof e?e:i,o[1]=s;for(var c=2;c<r;c++)o[c]=n[c];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},21750:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>m,frontMatter:()=>r,metadata:()=>s,toc:()=>c});var a=n(87462),i=(n(67294),n(3905));const r={sidebar_position:1.5},o="Getting the software",s={unversionedId:"genome_wide_association_studies/quantitative_trait_gwas/getting_the_software",id:"genome_wide_association_studies/quantitative_trait_gwas/getting_the_software",title:"Getting the software",description:"In this tutorial we will use two pieces of software designed for GWAS analysis:",source:"@site/docs/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software.md",sourceDirName:"genome_wide_association_studies/quantitative_trait_gwas",slug:"/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software.md",tags:[],version:"current",sidebarPosition:1.5,frontMatter:{sidebar_position:1.5},sidebar:"sidebar8",previous:{title:"Getting the data",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data"},next:{title:"Part 1: Quality control of genotype data",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes"}},l={},c=[],p={toc:c},d="wrapper";function m(e){let{components:t,...r}=e;return(0,i.kt)(d,(0,a.Z)({},p,r,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"getting-the-software"},"Getting the software"),(0,i.kt)("p",null,"In this tutorial we will use two pieces of software designed for GWAS analysis:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://www.cog-genomics.org/plink/"},(0,i.kt)("inlineCode",{parentName:"a"},"plink 1.9")," 'beta'"),", which is a general-purpose tool for manipulating genome-wide data"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://yanglab.westlake.edu.cn/software/gcta/"},(0,i.kt)("inlineCode",{parentName:"a"},"gcta"))," which implements (amongst other thins) a ",(0,i.kt)("strong",{parentName:"li"},"linear mixed model")," for association analysis.")),(0,i.kt)("p",null,"We will also R to make plots and (for one step) we'll write an exciting python script!"),(0,i.kt)("p",null,"To make things as easy as possible, we have included both ",(0,i.kt)("inlineCode",{parentName:"p"},"plink")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"gcta")," in the data tarball."),(0,i.kt)("p",null,"Look in the ",(0,i.kt)("inlineCode",{parentName:"p"},"software")," folder and pick the relevant version for your platform.  "),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Windows note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Windows users should work in the Ubuntu for Windows system, and use the linux version."))),(0,i.kt)("p",null,"For this tutorial we recommend copying the relevant versions into the top-level ",(0,i.kt)("inlineCode",{parentName:"p"},"gwas_tutorial")," folder"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"cp software/[version]/plink ./\ncp software/[version]/gcta64 ./\n")),(0,i.kt)("p",null,"You can then test the software out:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"./plink\n./gcta64\n")),(0,i.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"Mac firewall note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"On a mac, you may need to allow access to this software using System Preferences. This lives in ",(0,i.kt)("inlineCode",{parentName:"p"},"System Preferences")," ->\n",(0,i.kt)("inlineCode",{parentName:"p"},"Privacy and Security")," -> scroll down to 'Security' pane.  ",(0,i.kt)("strong",{parentName:"p"},"After")," you have tried to run the software once, you should\nsee a box like this:"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("img",{alt:"img",src:n(58133).Z,width:"483",height:"209"})),(0,i.kt)("p",{parentName:"div"},"If you give permission, you're able to run the software on subsequent attempts."))),(0,i.kt)("p",null,"If all works out, you should now see a message like:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'$ ./plink\nPLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)            cog-genomics.org/plink/1.9/\n(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n\n  plink <input flag(s)...> [command flag(s)...] [other flag(s)...]\n  plink --help [flag name(s)...]\n\nCommands include --make-bed, --recode, --flip-scan, --merge-list,\n--write-snplist, --list-duplicate-vars, --freqx, --missing, --test-mishap,\n--hardy, --mendel, --ibc, --impute-sex, --indep-pairphase, --r2, --show-tags,\n--blocks, --distance, --genome, --homozyg, --make-rel, --make-grm-gz,\n--rel-cutoff, --cluster, --pca, --neighbour, --ibs-test, --regress-distance,\n--model, --bd, --gxe, --logistic, --dosage, --lasso, --test-missing,\n--make-perm-pheno, --tdt, --qfam, --annotate, --clump, --gene-report,\n--meta-analysis, --epistasis, --fast-epistasis, and --score.\n\n"plink --help | more" describes all functions (warning: long).\n')),(0,i.kt)("p",null,"and"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"$ ./gcta64\n*******************************************************************\n* Genome-wide Complex Trait Analysis (GCTA)\n* version v1.94.1 Mac\n* (C) 2010-present, Yang Lab, Westlake University\n* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n*******************************************************************\nAnalysis started at 09:46:46 GMT on Wed Nov 06 2024.\nHostname: dhcp570.well.ox.ac.uk\n\nError: no analysis has been launched by the option(s)\nPlease see online documentation at https://yanglab.westlake.edu.cn/software/gcta/\n\n")),(0,i.kt)("p",null,"Congratulations!  You're ready to start the tutorial."))}m.isMDXComponent=!0},58133:(e,t,n)=>{n.d(t,{Z:()=>a});const a=n.p+"assets/images/mac_security-606dade5c640af45a9b1e38652555a38.png"}}]);