"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[4480],{3905:(e,t,a)=>{a.d(t,{Zo:()=>p,kt:()=>g});var n=a(67294);function s(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function i(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){s(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,s=function(e,t){if(null==e)return{};var a,n,s={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(s[a]=e[a]);return s}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(s[a]=e[a])}return s}var o=n.createContext({}),m=function(e){var t=n.useContext(o),a=t;return e&&(a="function"==typeof e?e(t):i(i({},t),e)),a},p=function(e){var t=m(e.components);return n.createElement(o.Provider,{value:t},e.children)},d="mdxType",c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},u=n.forwardRef((function(e,t){var a=e.components,s=e.mdxType,r=e.originalType,o=e.parentName,p=l(e,["components","mdxType","originalType","parentName"]),d=m(a),u=s,g=d["".concat(o,".").concat(u)]||d[u]||c[u]||r;return a?n.createElement(g,i(i({ref:t},p),{},{components:a})):n.createElement(g,i({ref:t},p))}));function g(e,t){var a=arguments,s=t&&t.mdxType;if("string"==typeof e||s){var r=a.length,i=new Array(r);i[0]=u;var l={};for(var o in t)hasOwnProperty.call(t,o)&&(l[o]=t[o]);l.originalType=e,l[d]="string"==typeof e?e:s,i[1]=l;for(var m=2;m<r;m++)i[m]=a[m];return n.createElement.apply(null,i)}return n.createElement.apply(null,a)}u.displayName="MDXCreateElement"},36127:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>o,contentTitle:()=>i,default:()=>c,frontMatter:()=>r,metadata:()=>l,toc:()=>m});var n=a(87462),s=(a(67294),a(3905));const r={sidebar_position:4},i="Part 1: Quality control of genotype data",l={unversionedId:"genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes",id:"genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes",title:"Part 1: Quality control of genotype data",description:"Calculating summary statistics for the data",source:"@site/docs/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes.md",sourceDirName:"genome_wide_association_studies/quantitative_trait_gwas",slug:"/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes.md",tags:[],version:"current",sidebarPosition:4,frontMatter:{sidebar_position:4},sidebar:"sidebar8",previous:{title:"Getting the software",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software"},next:{title:"Estimating relatedness",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/relatedness"}},o={},m=[{value:"Calculating summary statistics for the data",id:"calculating-summary-statistics-for-the-data",level:2},{value:"Missingness",id:"missingness",level:3},{value:"Heterozygosity",id:"heterozygosity",level:3},{value:"Applying filters to the data",id:"applying-filters-to-the-data",level:3}],p={toc:m},d="wrapper";function c(e){let{components:t,...a}=e;return(0,s.kt)(d,(0,n.Z)({},p,a,{components:t,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"part-1-quality-control-of-genotype-data"},"Part 1: Quality control of genotype data"),(0,s.kt)("h2",{id:"calculating-summary-statistics-for-the-data"},"Calculating summary statistics for the data"),(0,s.kt)("p",null,"The premise of our GWAS is that we have taken a random sample of case and control individuals in the population, and measured their genotypes genome-wide using a microarray approach.  Because of this there are a few quality control checks we ought to do before analysis - and, if there are any artifacts, we will want to remove them from the data before trying to test for association."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Why do quality control?")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Bear in mind that in a GWAS you are running association tests across millions of genetic variants at the same time. If\neven a small proportion of variants have data quality issues, this can easily lead to hundreds or thousands of spurious\nfindings.  You want to avoid this by getting rid of any obvious artifacts beforehand."),(0,s.kt)("p",{parentName:"div"},"If you are in any doubt about this, here are two cautionary tales:"),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Example 1"),": In 2011 a paper in Science reported a GWAS of ",(0,s.kt)("em",{parentName:"p"},"extreme longevity")," - purporting to find hundreds of loci across the genome which might make you live to a very old age.  However, within days it became clear that a substantial part of this signal was due to a quality control issue - different samples sets had been typed on different microarrays, and the quality control wasn't adequate. The paper was ",(0,s.kt)("a",{parentName:"p",href:"https://retractionwatch.com/2011/07/21/sebastiani-group-retracts-genetics-of-aging-study-from-science/"},"eventually retracted"),"."),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Example 2"),": In 2020 another study reported that the CCR-delta5 mutation ",(0,s.kt)("a",{parentName:"p",href:"https://doi.org/10.1038/s41591-019-0637-6"},"is deleterous")," (that is - it is associated with elevated mortality) in the UK Biobank.  This was also very high-profile because this was the gene at the centre of the creation of ",(0,s.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/He_Jiankui_affair"},"gene-edited children")," in 2019.  ",(0,s.kt)("strong",{parentName:"p"},"However")," - this apparent deleterous effect again turned out to be an ",(0,s.kt)("a",{parentName:"p",href:"https://doi.org/10.1038/s41591-019-0710-1"},"artifact of the genotyping quality")," (albeit this time, discovered and reported by the original study author).  In short this lead again to a ",(0,s.kt)("a",{parentName:"p",href:"https://www.nature.com/articles/s41591-019-0637-6"},"retraction"),"."),(0,s.kt)("p",{parentName:"div"},"The moral is: anything that ",(0,s.kt)("em",{parentName:"p"},"can")," go wrong with data, ",(0,s.kt)("em",{parentName:"p"},"will")," go wrong - you must do your QC carefully!"))),(0,s.kt)("p",null,"The simplest way to assess quality of genotyping data is by looking at the amount of missing genotype calls (or ",(0,s.kt)("strong",{parentName:"p"},"missingness"),") within the data.  This can be assessed at each variant or for each sample.  Variants with high missingness might mean the genotype assays for those variants are poor, while samples with high missingness might be ones that had low DNA quality or quantity, or other problems in collection."),(0,s.kt)("p",null,"Another useful way to inspect the data is to look at some genetic characteristics, such as the ",(0,s.kt)("strong",{parentName:"p"},"level of relatedness")," between individuals, the ",(0,s.kt)("strong",{parentName:"p"},"occurence of rare alleles"),", and the ",(0,s.kt)("strong",{parentName:"p"},"heterozygosity")," (proportion of genotype calls which are heterozygous) of each individual.  The heterozygosity in particular tells us useful information about the sample (for example, it might highlight samples that have been contaminated with other samples during processing - they would have too much heterozygosity)."),(0,s.kt)("p",null,"Let's start by using plink to calculating missingness.  First let's make a folder to put output in:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"mkdir output\n")),(0,s.kt)("p",null,"and compute missingness:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"./plink \\\n--bfile Genotype_data/AMR_genotypes \\\n--missing \\\n--out output/AMR_genotypes\n")),(0,s.kt)("p",null,"This will give you two files in your output directory:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("inlineCode",{parentName:"li"},"AMR_genotypes.imiss")," contains a summary of missing data by sample,"),(0,s.kt)("li",{parentName:"ul"},"and ",(0,s.kt)("inlineCode",{parentName:"li"},"AMR_genotypes.lmiss")," contains a similar summary by variant.")),(0,s.kt)("p",null,"Other summary statistics can be generated by replacing the --missing flag with other flags as shown below:"),(0,s.kt)("table",null,(0,s.kt)("thead",{parentName:"table"},(0,s.kt)("tr",{parentName:"thead"},(0,s.kt)("th",{parentName:"tr",align:null},"Descriptor"),(0,s.kt)("th",{parentName:"tr",align:null},"Flag"),(0,s.kt)("th",{parentName:"tr",align:null},"Statistic column"))),(0,s.kt)("tbody",{parentName:"table"},(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"missingness")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--missing")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"F_MISS"))),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"heterozygosity")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--het")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"OBS_HET")," (but see below.)")),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"relatedness")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--genome")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"PI_HAT")," (but see below.)")),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"minor allele frequency")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--freq")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"MAF"))),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"Hardy-Weinberg equilibrium")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--hardy")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"P"))))),(0,s.kt)("p",null,"Once you have generated summaries for each of these descriptors, you can plot these in R. "),(0,s.kt)("p",null,"For example:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},'library(tidyverse)\n\n# Read in the PLINK output file \ndata = readr::read_table( "output/AMR_genotypes.lmiss" )\n\np = (\n    ggplot( data, aes( x = F_MISS ) )\n    + geom_histogram( binwidth=0.001 )\n    + labs(\n        title = "Missingness (variant)",\n        x = "frequency",\n        y = "count"\n    )\n)\nprint(p)\nggsave( p, file = "output/AMR.missing_variants.png" )\n')),(0,s.kt)("p",null,"Here is a quick guide to each metric:"),(0,s.kt)("h3",{id:"missingness"},"Missingness"),(0,s.kt)("p",null,"'Missingess' is the proportion of missing genotypes.  Let's look at samples first:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},'miss = (\n    readr::read_table( "output/AMR_genotypes.imiss" )\n    %>% mutate(\n        missingness = F_MISS\n    )\n)\n')),(0,s.kt)("h3",{id:"heterozygosity"},"Heterozygosity"),(0,s.kt)("p",null,"The output file of ",(0,s.kt)("inlineCode",{parentName:"p"},"--het")," lists the observed number of homozygotes (",(0,s.kt)("inlineCode",{parentName:"p"},"O(HOM)"),") for each sample, and the total number of\nnon-missing genotypes (",(0,s.kt)("inlineCode",{parentName:"p"},"N(NM)"),")."),(0,s.kt)("p",null,"Let's use that along with the missingness to plot the rate of missing and heterozygous genotypes per sample:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},'het = (\n    readr::read_table( "output/AMR_genotypes.het" )\n    %>% mutate(\n        heterozygosity = (`N(NM)` - `O(HOM)`) / `N(NM)`\n    )\n)\nsample_summary = (\n    miss\n    %>% select( FID, IID, N_MISS, N_GENO, missingness )\n    %>% inner_join( het, by = c( "FID", "IID" ))\n)\nprint( sample_summary )\n\n')),(0,s.kt)("p",null,"...and plot it:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-r"},"(\n    ggplot( data = sample_summary )\n    + geom_point( aes( x = missingness, y = heterozygosity ))\n)\n")),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Make a plot like the above for each of the summary statistics."))),(0,s.kt)("h3",{id:"applying-filters-to-the-data"},"Applying filters to the data"),(0,s.kt)("p",null,"The next step will be to apply filters to the dataset using PLINK to exclude variants or samples of low quality. A description of the filters and some recommended thresholds is included below along with an example plink command. These flags can be used individually or together to apply filters to the dataset."),(0,s.kt)("table",null,(0,s.kt)("thead",{parentName:"table"},(0,s.kt)("tr",{parentName:"thead"},(0,s.kt)("th",{parentName:"tr",align:null},"Filter"),(0,s.kt)("th",{parentName:"tr",align:null},"Flag"),(0,s.kt)("th",{parentName:"tr",align:null},"Suggested cutoff"),(0,s.kt)("th",{parentName:"tr",align:null},"Description"))),(0,s.kt)("tbody",{parentName:"table"},(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},"missingness (variant)"),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--geno")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"0.1")),(0,s.kt)("td",{parentName:"tr",align:null},"Removes variants with",(0,s.kt)("br",null),"a ",(0,s.kt)("strong",{parentName:"td"},"missing call rate")," ",(0,s.kt)("span",{parentName:"td",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mo",{parentName:"mrow"},">"),(0,s.kt)("mn",{parentName:"mrow"},"0.1")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"> 0.1")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.5782em",verticalAlign:"-0.0391em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},">"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.1"))))))),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},"missingness (sample)"),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--mind")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"0.1")),(0,s.kt)("td",{parentName:"tr",align:null},"Removes samples with",(0,s.kt)("br",null),"a ",(0,s.kt)("strong",{parentName:"td"},"missing call rate")," ",(0,s.kt)("span",{parentName:"td",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mo",{parentName:"mrow"},">"),(0,s.kt)("mn",{parentName:"mrow"},"0.1")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"> 0.1")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.5782em",verticalAlign:"-0.0391em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},">"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.1"))))))),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},"minor allele frequency"),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--maf")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"0.01")),(0,s.kt)("td",{parentName:"tr",align:null},"Removes variants with",(0,s.kt)("br",null),"a ",(0,s.kt)("strong",{parentName:"td"},"minor allele frequency")," less than 1%")),(0,s.kt)("tr",{parentName:"tbody"},(0,s.kt)("td",{parentName:"tr",align:null},"Hardy-Weinberg equilibrium"),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"--hwe")),(0,s.kt)("td",{parentName:"tr",align:null},(0,s.kt)("inlineCode",{parentName:"td"},"0.000001")),(0,s.kt)("td",{parentName:"tr",align:null},"Removes variants which have ",(0,s.kt)("strong",{parentName:"td"},"Hardy-Weinberg",(0,s.kt)("br",null),'"exact test" p-value')," ",(0,s.kt)("span",{parentName:"td",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mo",{parentName:"mrow"},"<"),(0,s.kt)("mn",{parentName:"mrow"},"1"),(0,s.kt)("mo",{parentName:"mrow"},"\xd7"),(0,s.kt)("mn",{parentName:"mrow"},"1"),(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mn",{parentName:"msup"},"0"),(0,s.kt)("mrow",{parentName:"msup"},(0,s.kt)("mo",{parentName:"mrow"},"\u2212"),(0,s.kt)("mn",{parentName:"mrow"},"6")))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"< 1\\times 10^{-6}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.5782em",verticalAlign:"-0.0391em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},"<"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7278em",verticalAlign:"-0.0833em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"1"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\xd7"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"1"),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord"},"0"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"\u2212"),(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"6"))))))))))))))))),(0,s.kt)("p",null,"To filter variants and samples with genotyping rates below 0.1, and variants with a minor allele frequency less than 0.01, the following command would be used:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-bash"},"./plink \\\n--bfile Genotype_data/AMR_genotypes \\\n--geno 0.1 \\\n--mind 0.1 \\\n--maf 0.01 \\\n--make-bed \\\n--out Genotype_data/AMR_genotypes.filtered\n")),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Questions")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("ul",{parentName:"div"},(0,s.kt)("li",{parentName:"ul"},"Does this dataset consist of one chromosome or many? Which one(s)?"),(0,s.kt)("li",{parentName:"ul"},"How many individuals were in each dataset?"),(0,s.kt)("li",{parentName:"ul"},"How many individuals, and how many variants were excluded by the filters?")))))}c.isMDXComponent=!0}}]);