"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[7172],{3905:(e,t,n)=>{n.d(t,{Zo:()=>u,kt:()=>h});var r=n(67294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function s(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function i(e,t){if(null==e)return{};var n,r,a=function(e,t){if(null==e)return{};var n,r,a={},o=Object.keys(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var l=r.createContext({}),c=function(e){var t=r.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):s(s({},t),e)),n},u=function(e){var t=c(e.components);return r.createElement(l.Provider,{value:t},e.children)},m="mdxType",p={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},g=r.forwardRef((function(e,t){var n=e.components,a=e.mdxType,o=e.originalType,l=e.parentName,u=i(e,["components","mdxType","originalType","parentName"]),m=c(n),g=a,h=m["".concat(l,".").concat(g)]||m[g]||p[g]||o;return n?r.createElement(h,s(s({ref:t},u),{},{components:n})):r.createElement(h,s({ref:t},u))}));function h(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var o=n.length,s=new Array(o);s[0]=g;var i={};for(var l in t)hasOwnProperty.call(t,l)&&(i[l]=t[l]);i.originalType=e,i[m]="string"==typeof e?e:a,s[1]=i;for(var c=2;c<o;c++)s[c]=n[c];return r.createElement.apply(null,s)}return r.createElement.apply(null,n)}g.displayName="MDXCreateElement"},72360:(e,t,n)=>{n.d(t,{Z:()=>s});var r=n(67294),a=n(86010);const o={tabItem:"tabItem_OmH5"};function s(e){let{children:t,hidden:n,className:s}=e;return r.createElement("div",{role:"tabpanel",className:(0,a.Z)(o.tabItem,s),hidden:n},t)}},9877:(e,t,n)=>{n.d(t,{Z:()=>p});var r=n(87462),a=n(67294),o=n(72389),s=n(67392),i=n(7094),l=n(12466),c=n(86010);const u={tabList:"tabList_uSqn",tabItem:"tabItem_LplD"};function m(e){const{lazy:t,block:n,defaultValue:o,values:m,groupId:p,className:g}=e,h=a.Children.map(e.children,(e=>{if((0,a.isValidElement)(e)&&void 0!==e.props.value)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)})),f=m??h.map((e=>{let{props:{value:t,label:n,attributes:r}}=e;return{value:t,label:n,attributes:r}})),d=(0,s.l)(f,((e,t)=>e.value===t.value));if(d.length>0)throw new Error(`Docusaurus error: Duplicate values "${d.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`);const b=null===o?o:o??h.find((e=>e.props.default))?.props.value??h[0]?.props.value;if(null!==b&&!f.some((e=>e.value===b)))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${b}" but none of its children has the corresponding value. Available values are: ${f.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);const{tabGroupChoices:_,setTabGroupChoices:y}=(0,i.U)(),[v,w]=(0,a.useState)(b),k=[],{blockElementScrollPositionUntilNextRender:E}=(0,l.o5)();if(null!=p){const e=_[p];null!=e&&e!==v&&f.some((t=>t.value===e))&&w(e)}const O=e=>{const t=e.currentTarget,n=k.indexOf(t),r=f[n].value;r!==v&&(E(t),w(r),null!=p&&y(p,r))},N=e=>{let t=null;switch(e.key){case"ArrowRight":{const n=k.indexOf(e.currentTarget)+1;t=k[n]||k[0];break}case"ArrowLeft":{const n=k.indexOf(e.currentTarget)-1;t=k[n]||k[k.length-1];break}}t?.focus()};return a.createElement("div",{className:(0,c.Z)("tabs-container",u.tabList)},a.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,c.Z)("tabs",{"tabs--block":n},g)},f.map((e=>{let{value:t,label:n,attributes:o}=e;return a.createElement("li",(0,r.Z)({role:"tab",tabIndex:v===t?0:-1,"aria-selected":v===t,key:t,ref:e=>k.push(e),onKeyDown:N,onFocus:O,onClick:O},o,{className:(0,c.Z)("tabs__item",u.tabItem,o?.className,{"tabs__item--active":v===t})}),n??t)}))),t?(0,a.cloneElement)(h.filter((e=>e.props.value===v))[0],{className:"margin-top--md"}):a.createElement("div",{className:"margin-top--md"},h.map(((e,t)=>(0,a.cloneElement)(e,{key:t,hidden:e.props.value!==v})))))}function p(e){const t=(0,o.Z)();return a.createElement(m,(0,r.Z)({key:String(t)},e))}},66700:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>s,default:()=>p,frontMatter:()=>o,metadata:()=>i,toc:()=>c});var r=n(87462),a=(n(67294),n(3905));n(9877),n(72360);const o={sidebar_position:10},s="How much of the genome is in genes?",i={unversionedId:"bioinformatics/analysing_genes_across_species/how_much_in_genes/README",id:"bioinformatics/analysing_genes_across_species/how_much_in_genes/README",title:"How much of the genome is in genes?",description:"If you've followed so far you should have:",source:"@site/docs/bioinformatics/analysing_genes_across_species/how_much_in_genes/README.md",sourceDirName:"bioinformatics/analysing_genes_across_species/how_much_in_genes",slug:"/bioinformatics/analysing_genes_across_species/how_much_in_genes/",permalink:"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/bioinformatics/analysing_genes_across_species/how_much_in_genes/README.md",tags:[],version:"current",sidebarPosition:10,frontMatter:{sidebar_position:10},sidebar:"sidebar2",previous:{title:"Getting chromosome lengths",permalink:"/chg-training-resources/bioinformatics/analysing_genes_across_species/getting_chromosomes"},next:{title:"A naive approach",permalink:"/chg-training-resources/bioinformatics/analysing_genes_across_species/how_much_in_genes/naive_approach"}},l={},c=[],u={toc:c},m="wrapper";function p(e){let{components:t,...n}=e;return(0,a.kt)(m,(0,r.Z)({},u,n,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("h1",{id:"how-much-of-the-genome-is-in-genes"},"How much of the genome is in genes?"),(0,a.kt)("p",null,"If you've followed so far you should have:"),(0,a.kt)("ul",null,(0,a.kt)("li",{parentName:"ul"},"a table ",(0,a.kt)("inlineCode",{parentName:"li"},"genes")," of genes (with some statistics from the ",(0,a.kt)("a",{parentName:"li",href:"/chg-training-resources/bioinformatics/analysing_genes_across_species/extreme_genes/canonical_transcripts"},"'canonical' transcript"),")."),(0,a.kt)("li",{parentName:"ul"},"a table ",(0,a.kt)("inlineCode",{parentName:"li"},"regions")," of regions including chromosomes or other contigs, describing the assembly."),(0,a.kt)("li",{parentName:"ul"},"and you should have (or know how to get) other things like a dataframe of ",(0,a.kt)("inlineCode",{parentName:"li"},"transcripts")," and an ",(0,a.kt)("inlineCode",{parentName:"li"},"exons"),".")),(0,a.kt)("p",null,"Let's use this to try to figure out how much of the genome is actually covered by genes."))}p.isMDXComponent=!0}}]);