"use strict";(self.webpackChunkchg_training_resources=self.webpackChunkchg_training_resources||[]).push([[5972],{3905:(e,t,n)=>{n.d(t,{Zo:()=>c,kt:()=>m});var i=n(67294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,i)}return n}function s(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function o(e,t){if(null==e)return{};var n,i,a=function(e,t){if(null==e)return{};var n,i,a={},r=Object.keys(e);for(i=0;i<r.length;i++)n=r[i],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(i=0;i<r.length;i++)n=r[i],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var l=i.createContext({}),p=function(e){var t=i.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):s(s({},t),e)),n},c=function(e){var t=p(e.components);return i.createElement(l.Provider,{value:t},e.children)},u="mdxType",d={inlineCode:"code",wrapper:function(e){var t=e.children;return i.createElement(i.Fragment,{},t)}},g=i.forwardRef((function(e,t){var n=e.components,a=e.mdxType,r=e.originalType,l=e.parentName,c=o(e,["components","mdxType","originalType","parentName"]),u=p(n),g=a,m=u["".concat(l,".").concat(g)]||u[g]||d[g]||r;return n?i.createElement(m,s(s({ref:t},c),{},{components:n})):i.createElement(m,s({ref:t},c))}));function m(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var r=n.length,s=new Array(r);s[0]=g;var o={};for(var l in t)hasOwnProperty.call(t,l)&&(o[l]=t[l]);o.originalType=e,o[u]="string"==typeof e?e:a,s[1]=o;for(var p=2;p<r;p++)s[p]=n[p];return i.createElement.apply(null,s)}return i.createElement.apply(null,n)}g.displayName="MDXCreateElement"},40448:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>s,default:()=>d,frontMatter:()=>r,metadata:()=>o,toc:()=>p});var i=n(87462),a=(n(67294),n(3905));const r={sidebar_position:100},s="Appendix: Looking for independent signals",o={unversionedId:"genome_wide_association_studies/quantitative_trait_gwas/independent_signals",id:"genome_wide_association_studies/quantitative_trait_gwas/independent_signals",title:"Appendix: Looking for independent signals",description:"It is of course possible that there are multiple independent signals within a single region. To test if this is the",source:"@site/docs/genome_wide_association_studies/quantitative_trait_gwas/independent_signals.md",sourceDirName:"genome_wide_association_studies/quantitative_trait_gwas",slug:"/genome_wide_association_studies/quantitative_trait_gwas/independent_signals",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/independent_signals",draft:!1,editUrl:"https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/quantitative_trait_gwas/independent_signals.md",tags:[],version:"current",sidebarPosition:100,frontMatter:{sidebar_position:100},sidebar:"sidebar8",previous:{title:"Assessing effect sizes",permalink:"/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes"},next:{title:"Meta-analysis and fine-mapping tutorial",permalink:"/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/"}},l={},p=[],c={toc:p},u="wrapper";function d(e){let{components:t,...n}=e;return(0,a.kt)(u,(0,i.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("h1",{id:"appendix-looking-for-independent-signals"},"Appendix: Looking for independent signals"),(0,a.kt)("p",null,"It is of course possible that there are multiple independent signals within a single region. To test if this is the\ncase, we can carry out a conditional analysis on the region, using the allelic dosage of one SNP of interest (usually\nthe most significant). An example using our causal SNP within FUT2 is provided below."),(0,a.kt)("p",null,"First we need generate a list of SNP IDs that we want to condition on. For a single ID, this can be done with"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre"},'echo "rs601338:49206674:G:A" > output/rs601338.extract.list\n')),(0,a.kt)("p",null,"Then use this file with GCTA to extract the allelic dosage of this SNP:"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-sh"},"./gcta64 \\\n--bfile Genotype_data/AMR_imputed.filtered \\\n--extract output/rs601338.extract.list \\\n--recode \\\n--out output/rs601338\n")),(0,a.kt)("p",null,"This will produce and xmat file containing the dosage information. We then unzip this file"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre"},"gunzip -c output/rs601338.xmat.gz > output/rs601338.xmat\n")),(0,a.kt)("p",null,"...and use it as a quantitative covariate in GCTA"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre",className:"language-sh"},"./gcta64 \\\n--mlma \\\n--bfile Genotype_data/AMR_imputed.filtered \\\n--grm grm/AMR_genotypes.filtered \\\n--pheno Phenotype_data/AMR_phenotype.INT_transformed.txt \\\n--mpheno 2 \\\n--qcovar output/rs601338.xmat \\\n--out output/AMR_imputed.filtered.rs601338\n")),(0,a.kt)("p",null,"We can use the output of this to plot manhattan and Q-Q plots as before. Looking at the manhattan plot for this analysis, we can see that the previous signal is no longer present, suggesting that all the SNPs that reached significance in that part of the genome are linked with that signal. This is consistent with the results from our PLINK LD clump analysis that suggested that there were ~130 SNPs in a linked group around our FUT2 SNP."),(0,a.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,a.kt)("div",{parentName:"div",className:"admonition-heading"},(0,a.kt)("h5",{parentName:"div"},(0,a.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,a.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,a.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,a.kt)("div",{parentName:"div",className:"admonition-content"},(0,a.kt)("p",{parentName:"div"},"Another way to do this kind of analysis is to use a full fine-mapping approach, such as the one ",(0,a.kt)("a",{parentName:"p",href:"/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/"},"described here"),"."))))}d.isMDXComponent=!0}}]);