<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-genome_wide_association_studies/quantitative_trait_gwas/imputed_data">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Part 5: Imputation | The CHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://chg.ox.ac.uk//chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/imputed_data/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Part 5: Imputation | The CHG training resources"><meta data-rh="true" name="description" content="Genotype imputation allows us to test associations at ungenotyped markers, increase genotyping density for fine-mapping"><meta data-rh="true" property="og:description" content="Genotype imputation allows us to test associations at ungenotyped markers, increase genotyping density for fine-mapping"><link data-rh="true" rel="icon" href="/chg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://chg.ox.ac.uk//chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/imputed_data/"><link data-rh="true" rel="alternate" href="https://chg.ox.ac.uk//chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/imputed_data/" hreflang="en"><link data-rh="true" rel="alternate" href="https://chg.ox.ac.uk//chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/imputed_data/" hreflang="x-default"><link rel="stylesheet" href="/chg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/chg-training-resources/assets/js/runtime~main.2a3d8fab.js" as="script">
<link rel="preload" href="/chg-training-resources/assets/js/main.e5fe2fc6.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/chg-training-resources/"><div class="navbar__logo"><img src="/chg-training-resources/img/CHG_logo_RGB.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/chg-training-resources/img/CHG_logo_negative_RGB.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/chg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/chg-training/chg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/chg-training-resources/genome_wide_association_studies/">Genome-wide association analysis</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/chg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/">Genome-wide association study example: a case-control trait</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genome-wide association study example: a case-control trait&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/">Genome-wide association with quantitative antibody levels</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genome-wide association with quantitative antibody levels&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_data/">Getting the data</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/getting_the_software/">Getting the software</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/qc_genotypes/">Part 1: Quality control of genotype data</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/relatedness/">Estimating relatedness</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/sample_relatedness/">Aside: excluding related samples</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/phenotype_qc/">Part 2: Transforming the phenotype data</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/testing_for_association/">Part 3: Running a simple regression analysis</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/population_structure/">Part 4: Working with population substructure and multi-ethnic populations</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/imputed_data/">Part 5: Imputation</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/interpretation/">Part 6: Interpreting GWAS signals</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/effect_sizes/">Assessing effect sizes</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/independent_signals/">Appendix: Looking for independent signals</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/chg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/">Meta-analysis and fine-mapping tutorial</a><button aria-label="Toggle the collapsible sidebar category &#x27;Meta-analysis and fine-mapping tutorial&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/chg-training-resources/genome_wide_association_studies/file_formats/">GWAS file formats</a><button aria-label="Toggle the collapsible sidebar category &#x27;GWAS file formats&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/chg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/"><span itemprop="name">Genome-wide association with quantitative antibody levels</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Part 5: Imputation</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Part 5: Imputation</h1><p>Genotype imputation allows us to test associations at ungenotyped markers, increase genotyping density for fine-mapping
of GWAS signals, or combine datasets that were genotyped on different arrays. Genotyping arrays are limited by the
number of SNPs and variants that can be included in the array. Genotype imputation is a method by which we can use our
genotype data and additional reference data to statistically infer missing genotypes. This additional data could include
data from other genotype arrays, whole genome sequences, exome capture, etc. There are a number of publicly available
databases which could be used as reference data, such as the 1000 Genomes Project, the Haplotype Reference Consortium,
and the GNOMAD database.</p><p>At its core, imputation works on the basis on comparing observed genotypes for each sample with those present in a
reference panel then using intersecting markers to identify reference haplotypes that are most similar to each sample
and fill the gaps in the genotyped sample. The choice of reference population is important when imputing genotype data.
Different populations will have different frequencies of alleles, and varying LD across the genome, both of which can
affect the accuracy of imputation. Populations may also have novel variants that are not present elsewhere, that are not
likely to be successfully imputed if it is not present within the reference population. </p><p>We have provided imputed genotype data for each of the sample datasets in <code>Imputed_data/</code>. The imputed data is generated
from the genotype data that you have been working with above, and imputed using SHAPEIT and IMPUTE2 with the entire 1000
Genomes phase 3 dataset as the reference panel.</p><p>IMPUTE2 provides data in the <a href="/chg-training-resources/genome_wide_association_studies/file_formats/#oxford-format">Oxford format</a>. Genotype data is stored in <code>.gen</code> files
and the associated sample information is saved in the <code>.sample</code> file. GCTA does not support this format, so to run the
regression using a GRM, you will need to convert to &quot;best-guess&quot; phenotypes using PLINK then provide these files to
GCTA.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="qc-of-imputed-data">QC of imputed data<a class="hash-link" href="#qc-of-imputed-data" title="Direct link to heading">​</a></h2><p>Before running regression analyses with imputed data, it is good practice to check the quality of your imputation and
filter out low-quality data. Additional information about the imputation stats for each marker can be found in the <code>.info</code>
files. We are going to use this data to check the imputation quality for each population and create a list of poorly
imputed markers to exclude from the association tests. This file contains an &quot;info score&quot; for each marker representing
the observed statistical information associated with allele frequency estimates, and a &quot;certainty&quot; which represents the
average certainty of best-guess genotypes. More information about the file format can be found on the IMPUTE website.</p><p>First we are going to check the distribution of info scores for each imputed marker.  Here is an example script to do
this:</p><div class="language-r codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-r codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">library</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain">ggplot2</span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Read in the info file</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">data </span><span class="token operator" style="color:#393A34">&lt;-</span><span class="token plain"> readr</span><span class="token operator" style="color:#393A34">::</span><span class="token plain">read_table</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;Imputed_data/AMR_imputed.info&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Provide a filename for the filtered output</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">filtered_variants </span><span class="token operator" style="color:#393A34">&lt;-</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;Imputed_data/AMR_imputed.exclude.list&quot;</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">exclude_data </span><span class="token operator" style="color:#393A34">&lt;-</span><span class="token plain"> subset</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> data</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> info </span><span class="token operator" style="color:#393A34">&lt;=</span><span class="token plain"> </span><span class="token number" style="color:#36acaa">0.3</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">readr</span><span class="token operator" style="color:#393A34">::</span><span class="token plain">write_tsv</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    exclude_data</span><span class="token operator" style="color:#393A34">$</span><span class="token plain">rs_id</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    file </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> filtered_variants</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">p </span><span class="token operator" style="color:#393A34">&lt;-</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ggplot</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        data </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> data</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        mapping </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> aes</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> x </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> info </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token operator" style="color:#393A34">+</span><span class="token plain"> geom_histogram</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> binwidth </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> </span><span class="token number" style="color:#36acaa">0.01</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token operator" style="color:#393A34">+</span><span class="token plain"> labs</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> title</span><span class="token operator" style="color:#393A34">=</span><span class="token string" style="color:#e3116c">&quot;Imputation info score&quot;</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> x</span><span class="token operator" style="color:#393A34">=</span><span class="token string" style="color:#e3116c">&quot;info score&quot;</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> y</span><span class="token operator" style="color:#393A34">=</span><span class="token string" style="color:#e3116c">&quot;count&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">ggsave</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> p</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> file </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;AMR.info_score.png&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ```</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">You should see that a fair number of variants have </span><span class="token operator" style="color:#393A34">*</span><span class="token operator" style="color:#393A34">*</span><span class="token plain">low IMPUTE info scores</span><span class="token operator" style="color:#393A34">*</span><span class="token operator" style="color:#393A34">*</span><span class="token plain"> but many have high scores.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">For this analysis we&#x27;ve excluded variants with `info </span><span class="token operator" style="color:#393A34">&lt;</span><span class="token plain"> </span><span class="token number" style="color:#36acaa">0.3</span><span class="token plain">`.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">This script will have generated a list of markers </span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain">`AMR_imputed.exclude.list``</span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"> to exclude from downstream analyses. We</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">can then convert the gen file to PLINK format using the following command</span><span class="token operator" style="color:#393A34">:</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">```sh</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">.</span><span class="token operator" style="color:#393A34">/</span><span class="token plain">plink \</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token operator" style="color:#393A34">-</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">gen Imputed_data</span><span class="token operator" style="color:#393A34">/</span><span class="token plain">AMR_imputed.gen.gz \</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token operator" style="color:#393A34">-</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">sample Imputed_data</span><span class="token operator" style="color:#393A34">/</span><span class="token plain">AMR_imputed.sample \</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token operator" style="color:#393A34">-</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">exclude Imputed_data</span><span class="token operator" style="color:#393A34">/</span><span class="token plain">AMR_imputed.exclude.list \</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token operator" style="color:#393A34">-</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">oxford</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">single</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">chr </span><span class="token number" style="color:#36acaa">19</span><span class="token plain"> \</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token operator" style="color:#393A34">-</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">make</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">bed \</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token operator" style="color:#393A34">-</span><span class="token operator" style="color:#393A34">-</span><span class="token plain">out Imputed_data</span><span class="token operator" style="color:#393A34">/</span><span class="token plain">AMR_imputed.filtered</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Challenge</h5></div><div class="admonition-content"><p>Now go and run the association tests as above, using the filtered imputed data.</p><p>How do the associations change when using genotyped or imputed data?</p><p>What are some possible reasons for this?</p></div></div></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/chg-training/chg-training-resources/edit/main/docs/genome_wide_association_studies/quantitative_trait_gwas/imputed_data.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/population_structure/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Part 4: Working with population substructure and multi-ethnic populations</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/chg-training-resources/genome_wide_association_studies/quantitative_trait_gwas/interpretation/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Part 6: Interpreting GWAS signals</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#qc-of-imputed-data" class="table-of-contents__link toc-highlight">QC of imputed data</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/chg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/chg-training/chg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2025 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/chg-training-resources/assets/js/runtime~main.2a3d8fab.js"></script>
<script src="/chg-training-resources/assets/js/main.e5fe2fc6.js"></script>
</body>
</html>