<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-statistical_modelling/Hidden_markov_models/modelling_copy_number_variation">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Maximum likelihood copy number inference | The CHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://chg.ox.ac.uk//chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Maximum likelihood copy number inference | The CHG training resources"><meta data-rh="true" name="description" content="Remember that the &#x27;likelihood function&#x27; should compute the &quot;probability of the data given the parameters of interest&quot;."><meta data-rh="true" property="og:description" content="Remember that the &#x27;likelihood function&#x27; should compute the &quot;probability of the data given the parameters of interest&quot;."><link data-rh="true" rel="icon" href="/chg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://chg.ox.ac.uk//chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation/"><link data-rh="true" rel="alternate" href="https://chg.ox.ac.uk//chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation/" hreflang="en"><link data-rh="true" rel="alternate" href="https://chg.ox.ac.uk//chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation/" hreflang="x-default"><link rel="stylesheet" href="/chg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/chg-training-resources/assets/js/runtime~main.2a3d8fab.js" as="script">
<link rel="preload" href="/chg-training-resources/assets/js/main.e5fe2fc6.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/chg-training-resources/"><div class="navbar__logo"><img src="/chg-training-resources/img/CHG_logo_RGB.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/chg-training-resources/img/CHG_logo_negative_RGB.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/chg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/chg-training/chg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/chg-training-resources/statistical_modelling/">Statistical modelling</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/chg-training-resources/statistical_modelling/introduction/">From basics to Bayes</a><button aria-label="Toggle the collapsible sidebar category &#x27;From basics to Bayes&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/chg-training-resources/statistical_modelling/probability_cheatsheet/">Probability cheatsheet</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/chg-training-resources/statistical_modelling/distributions_cheatsheet/">Distributions cheatsheet</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/chg-training-resources/statistical_modelling/regression_modelling/">Modelling statistical association with regression</a><button aria-label="Toggle the collapsible sidebar category &#x27;Modelling statistical association with regression&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/">Glycophorin CNV calling tutorial</a><button aria-label="Toggle the collapsible sidebar category &#x27;Glycophorin CNV calling tutorial&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/">Warmup: examining read coverage in a region</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation/">Maximum likelihood copy number inference</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation_bayes/">Bayesian copy number inference</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm/">Modelling CNVs using a HMM</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/chg-training-resources/statistical_modelling/notes/computing_pvalues/">notes</a></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/chg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/"><span itemprop="name">Glycophorin CNV calling tutorial</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Maximum likelihood copy number inference</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Maximum likelihood copy number inference</h1><p>Remember that the &#x27;likelihood function&#x27; should compute the &quot;probability of the data given the parameters of interest&quot;.
In our case, the &#x27;parameters&#x27; are the <strong>unobserved copy number</strong> and the data is - you guessed it - the read coverage!
Let&#x27;s use our <a href="/chg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/">empirical modelling</a> to set up a likelhood now.</p><p>Our general model of the effect of CNVs on coverage is that coverage of a site with copy number <code>c</code> should be <code>c</code> times
as large as at a site with copy number 1 - plus noise.  (Humans are diploid so the copy number at most sites is 2).</p><p>If we think of sequence reads as being generated from each copy independently, a bit of thought shows that the same
relationship happens for the variance as well. So following our normal distribution, our model for copy number c would be that</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mtext>coverage</mtext><mo>∼</mo><mi mathvariant="script">N</mi><mrow><mo fence="true">(</mo><mi>c</mi><mo>×</mo><mtext>mean</mtext><mo separator="true">,</mo><mi>c</mi><mo>×</mo><mtext>variance</mtext><mo fence="true">)</mo></mrow></mrow><annotation encoding="application/x-tex">\text{coverage} \sim \mathcal{N}\left( c \times \text{mean}, c \times \text{variance} \right)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.625em;vertical-align:-0.1944em"></span><span class="mord text"><span class="mord">coverage</span></span><span class="mspace" style="margin-right:0.2778em"></span><span class="mrel">∼</span><span class="mspace" style="margin-right:0.2778em"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em"></span><span class="mord mathcal" style="margin-right:0.14736em">N</span><span class="mspace" style="margin-right:0.1667em"></span><span class="minner"><span class="mopen delimcenter" style="top:0em">(</span><span class="mord mathnormal">c</span><span class="mspace" style="margin-right:0.2222em"></span><span class="mbin">×</span><span class="mspace" style="margin-right:0.2222em"></span><span class="mord text"><span class="mord">mean</span></span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.1667em"></span><span class="mord mathnormal">c</span><span class="mspace" style="margin-right:0.2222em"></span><span class="mbin">×</span><span class="mspace" style="margin-right:0.2222em"></span><span class="mord text"><span class="mord">variance</span></span><span class="mclose delimcenter" style="top:0em">)</span></span></span></span></span></span></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>This is a good formula, isn&#x27;t ot?  It provides a direct way to link the true copy number, which we want to know, to the
coverage (accounting for noise in the coverage values), that might allow us to infer copy numbers.</p></div></div><p>Let&#x27;s try that now. To make it work:</p><ul><li>We will assume only copy numbers of 0 to 5 are possible (i.e. from a homozygous deletion up to a possible three extra copies).  Feel free to increase this if you want to!</li><li>We&#x27;ll put the results in a giant multidimensional array of <code>sites</code> x <code>samples</code> x <code>copy number states</code>.</li></ul><p>We start by computing the per-site likelihood of each data point for each possible copy number. Importantly <em>we will
continue to work throughout in log space</em> to avoid any numerical issues. This makes some computations more difficult,
but is a generally good idea when more than a few data points are involved.  So we&#x27;ll make our array <em>an array of
copy-number log-likelihoods</em> and call it <code>copy.number.lls</code>:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">copy.numbers = 0:5 </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">copy.number.lls = array(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    NA,                                 # fill with missing data to start</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dim = c(                            # dimensions</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        nrow  ( sites        ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        length( samples      ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        length( copy.numbers )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dimnames = list(                    # A good feature of R is you can name everything</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        sites$position,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        samples,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        sprintf( &quot;cf=%d&quot;, copy.numbers )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Let&#x27;s compute these lls:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">for( sample in 1:length(samples) ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    for( i in 1:length(copy.numbers) ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        copy.number = copy.numbers[i]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        mean.multiplier = copy.number/2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        variance.multiplier = max( copy.number/2, 0.01 ) # this to be error tolerant, as explained below</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        copy.number.lls[,sample,i] = gaussian.ll(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            X[,sample],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            params = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                mean = coverage.parameters$mean[sample] * mean.multiplier,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                variance = coverage.parameters$variance[sample] * variance.multiplier</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        # Handle the missing data sites.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        # We will just treat missing data as having likelihood 1 (loglikelihood zero) here.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        copy.number.lls[ is.na(X[,sample]),sample, ] = 0</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Look at the top left of the output:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">View( copy.number.lls[1:10,1:10,] )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>In practice it&#x27;s possible that even sites with zero &quot;real&quot; coverage might get some <em>observed</em> coverage. This could
happen due to spurious read alignment, or mis-alignment, for example.</p><p>For this reason we used a <code>variance.multiplier</code> variable above.  It equals <code>copy.number / 2</code> except for copy number
zero, where it allows a small amount of coverage to exist.</p></div></div><p>These copy number log-likelihoods should now capture some of the important signal in the data.  But how to plot it given it&#x27;s 3-dimensional?  Here are two ways we could do it.</p><p>We could take the maximum likelihood copy number call and plot that (for each sample and each site).</p><p>Or, we could recognise that the above is daft because it doesn&#x27;t take into account what we know - that most people will be diploid at most sites.  Instead, we could use a Bayesian approach to build this information in.</p><p>The rest of this practical does both these things.  We&#x27;ll use this function to plot the results:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.copy.numbers &lt;- function(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    copy.number,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    filename = NULL,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    title = NULL,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    palette = c( &quot;darkorange2&quot;, &quot;darkgoldenrod1&quot;, &quot;forestgreen&quot;, &quot;deepskyblue2&quot;, &quot;deepskyblue4&quot;, &quot;blueviolet&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">) { </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    if( !is.null( filename )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        pdf( file = filename, width = 6, height = 8 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    } </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    par(mfrow=c(1,1))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    par( mar = c( 4, 4, 2 + 2 *!is.null(title), 2 ) + 0.1 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    image( copy.number, x = 1:nrow(copy.number), y = 1:ncol(copy.number), xlab = &quot;sites&quot;, ylab = &quot;samples&quot;, col = palette, zlim = c( 0, length(palette)-1 ), main = title )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    legend( &quot;topleft&quot;, pch = 22, col = &#x27;black&#x27;, pt.bg = palette, legend = c( &quot;hom deletion&quot;, &quot;het deletion&quot;, &quot;normal&quot;, &quot;1 extra copy&quot;, &quot;2 extra copies&quot;, &quot;3 extra copies&quot; ), bg = &quot;white&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    if( !is.null( filename )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        dev.off()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="copy-number-inference-using-mles">Copy number inference using MLEs<a class="hash-link" href="#copy-number-inference-using-mles" title="Direct link to heading">​</a></h2><p>Let&#x27;s use our likelihood function to compute the <strong>maximum likelihood copy number state</strong> for each individual at each site:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">maximum.likelihood.state = array( NA, dim = c( nrow(X), ncol( X ) ), dimnames = list( sites$position, samples ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">for( variant in 1:nrow( X )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    for( sample in 1:ncol(X)) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        w = which.max( copy.number.lls[variant,sample,] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        stopifnot( !is.na(w) )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        maximum.likelihood.state[variant,sample] = copy.numbers[w]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.copy.numbers( maximum.likelihood.state, title = &quot;Maximum likelihood copy number&quot; )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>That plot definitely looks cleaner!  Still lots of noise though, can we do better?</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/chg-training/chg-training-resources/edit/main/docs/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Warmup: examining read coverage in a region</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/chg-training-resources/statistical_modelling/Hidden_markov_models/modelling_copy_number_variation_bayes/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Bayesian copy number inference</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#copy-number-inference-using-mles" class="table-of-contents__link toc-highlight">Copy number inference using MLEs</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/chg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/chg-training/chg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2025 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/chg-training-resources/assets/js/runtime~main.2a3d8fab.js"></script>
<script src="/chg-training-resources/assets/js/main.e5fe2fc6.js"></script>
</body>
</html>