---
sidebar_position: 4
---

# Analysing gene annotations in R or python

Welcome! This tutorial has two aims. On one hand it is about programming, specifically the type of
programming we often have to do in bioinformatics. This means handling large bits of data, making
sense of data, computing statistics and plotting. On the other hand, this tutorial aims to make you
familiar with an important type of genomic data: *gene annotation* files, what they contain and how
to interpret them.  The ultimate aim will be for you to put together an analysis of genes across a
range of species of interest.

:::tip Note

You can also  try the [command-line genes tutorial](/bioinformatics/exploring_genes_in_bash/)
tutorial first.

:::

To begin [go to the introduction](Introduction.md).

## Table of contents

1. [Introduction](001_Introduction.md)
2. [What gene annotation data looks like](002_What_gene_annotation_data_looks_like.md)
3. [Getting started writing some code](003_Getting_started_writing_some_code.md)
4. [Extracting attributes](004_extracting_attributes.md)
5. [Testing it out](005_testing_it_out.md)
6. [A quick recap (how did we do it?)](006_recap.md)
7. [Packaging up our code](007_making_a_module.md)
8. [Writing a useful command-line utility](008_Converting_gff_to_sqlite.md)
8. [Challenges](009_challenge_questions.md)
